From: Mangal Kushwah <mangalk2324@gmail.com>
Date: Sat, 31 Aug 2024 16:45:07 +0530
Subject: Use `vlc_fourcc_GetCodec` to map XVID to MP4V

---
 modules/demux/avi/avi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules/demux/avi/avi.c b/modules/demux/avi/avi.c
index bcc3511..a3204ea 100644
--- a/modules/demux/avi/avi.c
+++ b/modules/demux/avi/avi.c
@@ -713,7 +713,7 @@ static int Open( vlc_object_t * p_this )
                 }
                 else
                 {
-                    tk->fmt.i_codec = p_bih->biCompression;
+                    tk->fmt.i_codec = vlc_fourcc_GetCodec(VIDEO_ES, p_bih->biCompression);
                     if( tk->fmt.i_codec == VLC_CODEC_MP4V &&
                         !strncasecmp( (char*)&p_strh->i_handler, "XVID", 4 ) )
                     {
