Index: vlc-0.8.6-svn20061012.debian/modules/gui/wxwidgets/interface.cpp
===================================================================
--- vlc-0.8.6-svn20061012.debian.orig/modules/gui/wxwidgets/interface.cpp	2006-09-18 12:28:21.000000000 +0200
+++ vlc-0.8.6-svn20061012.debian/modules/gui/wxwidgets/interface.cpp	2007-01-08 02:25:50.000000000 +0100
@@ -606,8 +606,8 @@
     /* Create the "Help" menu */
     wxMenu *help_menu = new wxMenu;
     help_menu->Append( About_Event, wxU(_("About...")) );
-    help_menu->AppendSeparator();
-    help_menu->Append( UpdateVLC_Event, wxU(_("Check for Updates...")) );
+//    help_menu->AppendSeparator();
+//    help_menu->Append( UpdateVLC_Event, wxU(_("Check for Updates...")) );
 
     /* Append the freshly created menus to the menu bar... */
     wxMenuBar *menubar = new wxMenuBar();
