--- a/configure.ac	2008-08-22 18:57:54.000000000 +0200
+++ b/configure.ac	2008-08-22 19:02:49.000000000 +0200
@@ -5155,9 +5155,9 @@
   if test "${with_mozilla_sdk_path}" = "" -o "${with_mozilla_sdk_path}" = "no"
   then
     dnl New firefox-1.5 SDK api
-    PKG_CHECK_MODULES([FIREFOX], [firefox-plugin],
+    PKG_CHECK_MODULES([FIREFOX], [iceape-plugin],
       [
-        PKG_CHECK_MODULES([FIREFOX_XPCOM], [firefox-xpcom],
+        PKG_CHECK_MODULES([FIREFOX_XPCOM], [iceape-xpcom],
           [],
           [
             PKG_CHECK_MODULES([MOZILLA_NSPR], [mozilla-nspr],
@@ -5186,7 +5186,7 @@
         fi
         MOZILLA_REQUIRED_HEADERS=
         mozilla=:
-        firefox_include=`pkg-config --variable=includedir firefox-plugin`
+        firefox_include=`pkg-config --variable=includedir iceape-plugin`
         if test -f "${firefox_include}/mozilla-config.h" -a -f "${firefox_include}/mozilla-config32.h"; then
             MOZILLA_CONFIG_H="${firefox_include}/mozilla-config32.h"
         elif test -f "${firefox_include}/mozilla-config.h" -a -f "${firefox_include}/mozilla-config64.h"; then
