From: Steve Lhomme <robux4@ycbcr.xyz>
Date: Thu, 5 Dec 2024 10:50:50 +0100
Subject: qt: remove unused meta_to_mlmeta()

---
 modules/gui/qt/components/playlist/sorting.h | 22 ----------------------
 1 file changed, 22 deletions(-)

diff --git a/modules/gui/qt/components/playlist/sorting.h b/modules/gui/qt/components/playlist/sorting.h
index 6c1e935..64ce5e0 100644
--- a/modules/gui/qt/components/playlist/sorting.h
+++ b/modules/gui/qt/components/playlist/sorting.h
@@ -136,26 +136,4 @@ static inline int i_column_sorting( uint32_t i_column )
     }
 }
 
-/* Return the media library query select type */
-static inline ml_select_e meta_to_mlmeta( uint32_t i_column )
-{
-    switch( i_column )
-    {
-    case COLUMN_NUMBER:         return ML_ID;
-    case COLUMN_TITLE:          return ML_TITLE;
-    case COLUMN_DURATION:       return ML_DURATION;
-    case COLUMN_ARTIST:         return ML_ARTIST;
-    case COLUMN_GENRE:          return ML_GENRE;
-    case COLUMN_ALBUM:          return ML_ALBUM;
-    case COLUMN_TRACK_NUMBER:   return ML_TRACK_NUMBER;
-    case COLUMN_DESCRIPTION:    return ML_EXTRA;
-    case COLUMN_URI:            return ML_URI;
-    case COLUMN_RATING:         return ML_VOTE;
-    case COLUMN_COVER:          return ML_COVER;
-    case COLUMN_DISC_NUMBER:    return ML_DISC_NUMBER;
-    case COLUMN_DATE:           return ML_YEAR;
-    default: abort();
-    }
-}
-
 #endif
