From: Steve Lhomme <robux4@ycbcr.xyz>
Date: Tue, 17 Dec 2024 09:17:00 +0100
Subject: qt: add missing QActionGroup include

To be able to use rendererGroup.
---
 modules/gui/qt/actions_manager.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/modules/gui/qt/actions_manager.cpp b/modules/gui/qt/actions_manager.cpp
index efa3044..f1ada88 100644
--- a/modules/gui/qt/actions_manager.cpp
+++ b/modules/gui/qt/actions_manager.cpp
@@ -38,6 +38,8 @@
 #include "components/extended_panels.hpp"
 #include "menus.hpp"
 
+#include <QActionGroup>
+
 ActionsManager::ActionsManager( intf_thread_t * _p_i )
     : p_intf( _p_i )
     , m_scanning( false )
