Index: vlc-0.8.6.debian/modules/misc/network/ipv6.c
===================================================================
--- vlc-0.8.6.debian.orig/modules/misc/network/ipv6.c	2006-12-11 12:32:22.000000000 +0100
+++ vlc-0.8.6.debian/modules/misc/network/ipv6.c	2006-12-11 12:32:38.000000000 +0100
@@ -57,6 +57,7 @@
 #if defined (WIN32) && !defined (MCAST_JOIN_SOURCE_GROUP)
 /* Interim Vista definitions */
 #  define MCAST_JOIN_SOURCE_GROUP 45 /* from <ws2ipdef.h> */
+#if defined __GLIBC__ && !defined __FreeBSD_kernel__
 struct group_source_req
 {
        uint32_t           gsr_interface;  /* interface index */
@@ -64,6 +65,7 @@
        struct sockaddr_storage gsr_source;     /* source address */
 };
 #endif
+#endif
 
 /*****************************************************************************
  * Local prototypes
