Index: vlc-0.8.6-svn20061012.debian/modules/misc/network/ipv6.c
===================================================================
--- vlc-0.8.6-svn20061012.debian.orig/modules/misc/network/ipv6.c	2006-09-18 12:28:25.000000000 +0200
+++ vlc-0.8.6-svn20061012.debian/modules/misc/network/ipv6.c	2007-01-11 16:17:16.000000000 +0100
@@ -61,6 +61,7 @@
 # else
 #  define MCAST_JOIN_SOURCE_GROUP 46
 # endif
+#if defined __GLIBC__ && !defined __FreeBSD_kernel__
 struct group_source_req
 {
        uint32_t           gsr_interface;  /* interface index */
@@ -68,6 +69,7 @@
        struct sockaddr_storage gsr_source;     /* source address */
 };
 #endif
+#endif
 
 /*****************************************************************************
  * Local prototypes
