--- a/configure
+++ b/configure
@@ -988,13 +988,13 @@ HAS_WIDL_COMPILER_TRUE
 HAS_WIDL_COMPILER_FALSE
 BUILD_ACTIVEX_TRUE
 BUILD_ACTIVEX_FALSE
-FIREFOX_CFLAGS
-FIREFOX_LIBS
+ICEAPE_CFLAGS
+ICEAPE_LIBS
 MOZILLA_CONFIG
 SEAMONKEY_CONFIG
 XULRUNNER_CONFIG
-FIREFOX_XPCOM_CFLAGS
-FIREFOX_XPCOM_LIBS
+ICEAPE_XPCOM_CFLAGS
+ICEAPE_XPCOM_LIBS
 MOZILLA_NSPR_CFLAGS
 MOZILLA_NSPR_LIBS
 NSPR_CFLAGS
@@ -1091,10 +1091,10 @@ GNUTLS_CFLAGS
 GNUTLS_LIBS
 MIDL
 WIDL
-FIREFOX_CFLAGS
-FIREFOX_LIBS
-FIREFOX_XPCOM_CFLAGS
-FIREFOX_XPCOM_LIBS
+ICEAPE_CFLAGS
+ICEAPE_LIBS
+ICEAPE_XPCOM_CFLAGS
+ICEAPE_XPCOM_LIBS
 MOZILLA_NSPR_CFLAGS
 MOZILLA_NSPR_LIBS
 NSPR_CFLAGS
@@ -1973,14 +1973,13 @@ Some influential environment variables:
   GNUTLS_LIBS linker flags for GNUTLS, overriding pkg-config
   MIDL        Microsoft IDL compiler (Win32 platform only)
   WIDL        Wine IDL compiler (requires Wine SDK)
-  FIREFOX_CFLAGS
-              C compiler flags for FIREFOX, overriding pkg-config
-  FIREFOX_LIBS
-              linker flags for FIREFOX, overriding pkg-config
-  FIREFOX_XPCOM_CFLAGS
-              C compiler flags for FIREFOX_XPCOM, overriding pkg-config
-  FIREFOX_XPCOM_LIBS
-              linker flags for FIREFOX_XPCOM, overriding pkg-config
+  ICEAPE_CFLAGS
+              C compiler flags for ICEAPE, overriding pkg-config
+  ICEAPE_LIBS linker flags for ICEAPE, overriding pkg-config
+  ICEAPE_XPCOM_CFLAGS
+              C compiler flags for ICEAPE_XPCOM, overriding pkg-config
+  ICEAPE_XPCOM_LIBS
+              linker flags for ICEAPE_XPCOM, overriding pkg-config
   MOZILLA_NSPR_CFLAGS
               C compiler flags for MOZILLA_NSPR, overriding pkg-config
   MOZILLA_NSPR_LIBS
@@ -3023,11 +3022,6 @@ ac_config_headers="$ac_config_headers co
 
 
 
-if test "x${enable_maintainer_mode}" != "xno"; then
-  enable_maintainer_mode="yes"
-fi
-
-
 { echo "$as_me:$LINENO: checking whether to enable maintainer-specific portions of Makefiles" >&5
 echo $ECHO_N "checking whether to enable maintainer-specific portions of Makefiles... $ECHO_C" >&6; }
     # Check whether --enable-maintainer-mode was given.
@@ -6577,7 +6571,7 @@ ia64-*-hpux*)
   ;;
 *-*-irix6*)
   # Find out which ABI we are using.
-  echo '#line 6580 "configure"' > conftest.$ac_ext
+  echo '#line 6574 "configure"' > conftest.$ac_ext
   if { (eval echo "$as_me:$LINENO: \"$ac_compile\"") >&5
   (eval $ac_compile) 2>&5
   ac_status=$?
@@ -9216,11 +9210,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:9219: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:9213: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>conftest.err)
    ac_status=$?
    cat conftest.err >&5
-   echo "$as_me:9223: \$? = $ac_status" >&5
+   echo "$as_me:9217: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s "$ac_outfile"; then
      # The compiler can only warn and ignore the option if not recognized
      # So say no if there are warnings other than the usual output.
@@ -9506,11 +9500,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:9509: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:9503: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>conftest.err)
    ac_status=$?
    cat conftest.err >&5
-   echo "$as_me:9513: \$? = $ac_status" >&5
+   echo "$as_me:9507: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s "$ac_outfile"; then
      # The compiler can only warn and ignore the option if not recognized
      # So say no if there are warnings other than the usual output.
@@ -9610,11 +9604,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:9613: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:9607: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>out/conftest.err)
    ac_status=$?
    cat out/conftest.err >&5
-   echo "$as_me:9617: \$? = $ac_status" >&5
+   echo "$as_me:9611: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s out/conftest2.$ac_objext
    then
      # The compiler can only warn and ignore the option if not recognized
@@ -11987,7 +11981,7 @@ else
   lt_dlunknown=0; lt_dlno_uscore=1; lt_dlneed_uscore=2
   lt_status=$lt_dlunknown
   cat > conftest.$ac_ext <<EOF
-#line 11990 "configure"
+#line 11984 "configure"
 #include "confdefs.h"
 
 #if HAVE_DLFCN_H
@@ -12087,7 +12081,7 @@ else
   lt_dlunknown=0; lt_dlno_uscore=1; lt_dlneed_uscore=2
   lt_status=$lt_dlunknown
   cat > conftest.$ac_ext <<EOF
-#line 12090 "configure"
+#line 12084 "configure"
 #include "confdefs.h"
 
 #if HAVE_DLFCN_H
@@ -14488,11 +14482,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:14491: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:14485: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>conftest.err)
    ac_status=$?
    cat conftest.err >&5
-   echo "$as_me:14495: \$? = $ac_status" >&5
+   echo "$as_me:14489: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s "$ac_outfile"; then
      # The compiler can only warn and ignore the option if not recognized
      # So say no if there are warnings other than the usual output.
@@ -14592,11 +14586,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:14595: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:14589: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>out/conftest.err)
    ac_status=$?
    cat out/conftest.err >&5
-   echo "$as_me:14599: \$? = $ac_status" >&5
+   echo "$as_me:14593: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s out/conftest2.$ac_objext
    then
      # The compiler can only warn and ignore the option if not recognized
@@ -16190,11 +16184,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:16193: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:16187: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>conftest.err)
    ac_status=$?
    cat conftest.err >&5
-   echo "$as_me:16197: \$? = $ac_status" >&5
+   echo "$as_me:16191: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s "$ac_outfile"; then
      # The compiler can only warn and ignore the option if not recognized
      # So say no if there are warnings other than the usual output.
@@ -16294,11 +16288,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:16297: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:16291: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>out/conftest.err)
    ac_status=$?
    cat out/conftest.err >&5
-   echo "$as_me:16301: \$? = $ac_status" >&5
+   echo "$as_me:16295: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s out/conftest2.$ac_objext
    then
      # The compiler can only warn and ignore the option if not recognized
@@ -18514,11 +18508,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:18517: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:18511: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>conftest.err)
    ac_status=$?
    cat conftest.err >&5
-   echo "$as_me:18521: \$? = $ac_status" >&5
+   echo "$as_me:18515: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s "$ac_outfile"; then
      # The compiler can only warn and ignore the option if not recognized
      # So say no if there are warnings other than the usual output.
@@ -18804,11 +18798,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:18807: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:18801: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>conftest.err)
    ac_status=$?
    cat conftest.err >&5
-   echo "$as_me:18811: \$? = $ac_status" >&5
+   echo "$as_me:18805: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s "$ac_outfile"; then
      # The compiler can only warn and ignore the option if not recognized
      # So say no if there are warnings other than the usual output.
@@ -18908,11 +18902,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:18911: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:18905: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>out/conftest.err)
    ac_status=$?
    cat out/conftest.err >&5
-   echo "$as_me:18915: \$? = $ac_status" >&5
+   echo "$as_me:18909: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s out/conftest2.$ac_objext
    then
      # The compiler can only warn and ignore the option if not recognized
@@ -62427,20 +62421,20 @@ then
   then
 
 pkg_failed=no
-{ echo "$as_me:$LINENO: checking for FIREFOX" >&5
-echo $ECHO_N "checking for FIREFOX... $ECHO_C" >&6; }
+{ echo "$as_me:$LINENO: checking for ICEAPE" >&5
+echo $ECHO_N "checking for ICEAPE... $ECHO_C" >&6; }
 
 if test -n "$PKG_CONFIG"; then
-    if test -n "$FIREFOX_CFLAGS"; then
-        pkg_cv_FIREFOX_CFLAGS="$FIREFOX_CFLAGS"
+    if test -n "$ICEAPE_CFLAGS"; then
+        pkg_cv_ICEAPE_CFLAGS="$ICEAPE_CFLAGS"
     else
         if test -n "$PKG_CONFIG" && \
-    { (echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"firefox-plugin\"") >&5
-  ($PKG_CONFIG --exists --print-errors "firefox-plugin") 2>&5
+    { (echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"iceape-plugin\"") >&5
+  ($PKG_CONFIG --exists --print-errors "iceape-plugin") 2>&5
   ac_status=$?
   echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; then
-  pkg_cv_FIREFOX_CFLAGS=`$PKG_CONFIG --cflags "firefox-plugin" 2>/dev/null`
+  pkg_cv_ICEAPE_CFLAGS=`$PKG_CONFIG --cflags "iceape-plugin" 2>/dev/null`
 else
   pkg_failed=yes
 fi
@@ -62449,16 +62443,16 @@ else
 	pkg_failed=untried
 fi
 if test -n "$PKG_CONFIG"; then
-    if test -n "$FIREFOX_LIBS"; then
-        pkg_cv_FIREFOX_LIBS="$FIREFOX_LIBS"
+    if test -n "$ICEAPE_LIBS"; then
+        pkg_cv_ICEAPE_LIBS="$ICEAPE_LIBS"
     else
         if test -n "$PKG_CONFIG" && \
-    { (echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"firefox-plugin\"") >&5
-  ($PKG_CONFIG --exists --print-errors "firefox-plugin") 2>&5
+    { (echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"iceape-plugin\"") >&5
+  ($PKG_CONFIG --exists --print-errors "iceape-plugin") 2>&5
   ac_status=$?
   echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; then
-  pkg_cv_FIREFOX_LIBS=`$PKG_CONFIG --libs "firefox-plugin" 2>/dev/null`
+  pkg_cv_ICEAPE_LIBS=`$PKG_CONFIG --libs "iceape-plugin" 2>/dev/null`
 else
   pkg_failed=yes
 fi
@@ -62477,12 +62471,12 @@ else
         _pkg_short_errors_supported=no
 fi
         if test $_pkg_short_errors_supported = yes; then
-	        FIREFOX_PKG_ERRORS=`$PKG_CONFIG --short-errors --errors-to-stdout --print-errors "firefox-plugin"`
+	        ICEAPE_PKG_ERRORS=`$PKG_CONFIG --short-errors --errors-to-stdout --print-errors "iceape-plugin"`
         else
-	        FIREFOX_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "firefox-plugin"`
+	        ICEAPE_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "iceape-plugin"`
         fi
 	# Put the nasty error message in config.log where it belongs
-	echo "$FIREFOX_PKG_ERRORS" >&5
+	echo "$ICEAPE_PKG_ERRORS" >&5
 
 	{ echo "$as_me:$LINENO: result: no" >&5
 echo "${ECHO_T}no" >&6; }
@@ -62774,27 +62768,27 @@ echo "$as_me: error: Please install the 
 
 
 else
-	FIREFOX_CFLAGS=$pkg_cv_FIREFOX_CFLAGS
-	FIREFOX_LIBS=$pkg_cv_FIREFOX_LIBS
+	ICEAPE_CFLAGS=$pkg_cv_ICEAPE_CFLAGS
+	ICEAPE_LIBS=$pkg_cv_ICEAPE_LIBS
         { echo "$as_me:$LINENO: result: yes" >&5
 echo "${ECHO_T}yes" >&6; }
 
 
 pkg_failed=no
-{ echo "$as_me:$LINENO: checking for FIREFOX_XPCOM" >&5
-echo $ECHO_N "checking for FIREFOX_XPCOM... $ECHO_C" >&6; }
+{ echo "$as_me:$LINENO: checking for ICEAPE_XPCOM" >&5
+echo $ECHO_N "checking for ICEAPE_XPCOM... $ECHO_C" >&6; }
 
 if test -n "$PKG_CONFIG"; then
-    if test -n "$FIREFOX_XPCOM_CFLAGS"; then
-        pkg_cv_FIREFOX_XPCOM_CFLAGS="$FIREFOX_XPCOM_CFLAGS"
+    if test -n "$ICEAPE_XPCOM_CFLAGS"; then
+        pkg_cv_ICEAPE_XPCOM_CFLAGS="$ICEAPE_XPCOM_CFLAGS"
     else
         if test -n "$PKG_CONFIG" && \
-    { (echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"firefox-xpcom\"") >&5
-  ($PKG_CONFIG --exists --print-errors "firefox-xpcom") 2>&5
+    { (echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"iceape-xpcom\"") >&5
+  ($PKG_CONFIG --exists --print-errors "iceape-xpcom") 2>&5
   ac_status=$?
   echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; then
-  pkg_cv_FIREFOX_XPCOM_CFLAGS=`$PKG_CONFIG --cflags "firefox-xpcom" 2>/dev/null`
+  pkg_cv_ICEAPE_XPCOM_CFLAGS=`$PKG_CONFIG --cflags "iceape-xpcom" 2>/dev/null`
 else
   pkg_failed=yes
 fi
@@ -62803,16 +62797,16 @@ else
 	pkg_failed=untried
 fi
 if test -n "$PKG_CONFIG"; then
-    if test -n "$FIREFOX_XPCOM_LIBS"; then
-        pkg_cv_FIREFOX_XPCOM_LIBS="$FIREFOX_XPCOM_LIBS"
+    if test -n "$ICEAPE_XPCOM_LIBS"; then
+        pkg_cv_ICEAPE_XPCOM_LIBS="$ICEAPE_XPCOM_LIBS"
     else
         if test -n "$PKG_CONFIG" && \
-    { (echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"firefox-xpcom\"") >&5
-  ($PKG_CONFIG --exists --print-errors "firefox-xpcom") 2>&5
+    { (echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"iceape-xpcom\"") >&5
+  ($PKG_CONFIG --exists --print-errors "iceape-xpcom") 2>&5
   ac_status=$?
   echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; then
-  pkg_cv_FIREFOX_XPCOM_LIBS=`$PKG_CONFIG --libs "firefox-xpcom" 2>/dev/null`
+  pkg_cv_ICEAPE_XPCOM_LIBS=`$PKG_CONFIG --libs "iceape-xpcom" 2>/dev/null`
 else
   pkg_failed=yes
 fi
@@ -62831,12 +62825,12 @@ else
         _pkg_short_errors_supported=no
 fi
         if test $_pkg_short_errors_supported = yes; then
-	        FIREFOX_XPCOM_PKG_ERRORS=`$PKG_CONFIG --short-errors --errors-to-stdout --print-errors "firefox-xpcom"`
+	        ICEAPE_XPCOM_PKG_ERRORS=`$PKG_CONFIG --short-errors --errors-to-stdout --print-errors "iceape-xpcom"`
         else
-	        FIREFOX_XPCOM_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "firefox-xpcom"`
+	        ICEAPE_XPCOM_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "iceape-xpcom"`
         fi
 	# Put the nasty error message in config.log where it belongs
-	echo "$FIREFOX_XPCOM_PKG_ERRORS" >&5
+	echo "$ICEAPE_XPCOM_PKG_ERRORS" >&5
 
 	{ echo "$as_me:$LINENO: result: no" >&5
 echo "${ECHO_T}no" >&6; }
@@ -63320,13 +63314,13 @@ echo "${ECHO_T}yes" >&6; }
 fi
 
 else
-	FIREFOX_XPCOM_CFLAGS=$pkg_cv_FIREFOX_XPCOM_CFLAGS
-	FIREFOX_XPCOM_LIBS=$pkg_cv_FIREFOX_XPCOM_LIBS
+	ICEAPE_XPCOM_CFLAGS=$pkg_cv_ICEAPE_XPCOM_CFLAGS
+	ICEAPE_XPCOM_LIBS=$pkg_cv_ICEAPE_XPCOM_LIBS
         { echo "$as_me:$LINENO: result: yes" >&5
 echo "${ECHO_T}yes" >&6; }
 	:
 fi
-        CPPFLAGS="${CPPFLAGS_save} ${FIREFOX_CFLAGS} ${FIREFOX_XPCOM_CFLAGS} ${MOZILLA_NSPR_CFLAGS} ${NSPR_CFLAGS} ${MOZILLA_CPPFLAGS}"
+        CPPFLAGS="${CPPFLAGS_save} ${ICEAPE_CFLAGS} ${ICEAPE_XPCOM_CFLAGS} ${MOZILLA_NSPR_CFLAGS} ${NSPR_CFLAGS} ${MOZILLA_CPPFLAGS}"
         MOZILLA_REQUIRED_HEADERS=1
 
 for ac_header in mozilla-config.h
@@ -63682,7 +63676,7 @@ echo "$as_me: error: Please install the 
         fi
         MOZILLA_REQUIRED_HEADERS=
         mozilla=:
-        firefox_include=`pkg-config --variable=includedir firefox-plugin`
+        firefox_include=`pkg-config --variable=includedir iceape-plugin`
         if test -f "${firefox_include}/mozilla-config.h" -a -f "${firefox_include}/mozilla-config32.h"; then
             MOZILLA_CONFIG_H="${firefox_include}/mozilla-config32.h"
         elif test -f "${firefox_include}/mozilla-config.h" -a -f "${firefox_include}/mozilla-config64.h"; then
@@ -63706,7 +63700,7 @@ echo "$as_me: error: Please install the 
 
 
   for element in mozilla; do
-    eval "LDFLAGS_${element}="'"'"${FIREFOX_LIBS} ${MOZILLA_NSPR_LIBS} ${NSPR_LIBS} ${MOZILLA_LDFLAGS} "'$'"{LDFLAGS_${element}} "'"'
+    eval "LDFLAGS_${element}="'"'"${ICEAPE_LIBS} ${MOZILLA_NSPR_LIBS} ${NSPR_LIBS} ${MOZILLA_LDFLAGS} "'$'"{LDFLAGS_${element}} "'"'
     am_modules_with_ldflags="${am_modules_with_ldflags} ${element}"
   done
 
@@ -64829,7 +64823,7 @@ else
 JAVA_TEST=Test.java
 CLASS_TEST=Test.class
 cat << \EOF > $JAVA_TEST
-/* #line 64832 "configure" */
+/* #line 64826 "configure" */
 public class Test {
 }
 EOF
@@ -65005,7 +64999,7 @@ EOF
 if uudecode$EXEEXT Test.uue; then
         ac_cv_prog_uudecode_base64=yes
 else
-        echo "configure: 65008: uudecode had trouble decoding base 64 file 'Test.uue'" >&5
+        echo "configure: 65002: uudecode had trouble decoding base 64 file 'Test.uue'" >&5
         echo "configure: failed file was:" >&5
         cat Test.uue >&5
         ac_cv_prog_uudecode_base64=no
@@ -65127,7 +65121,7 @@ else
 JAVA_TEST=Test.java
 CLASS_TEST=Test.class
 cat << \EOF > $JAVA_TEST
-/* #line 65130 "configure" */
+/* #line 65124 "configure" */
 public class Test {
 }
 EOF
@@ -65164,7 +65158,7 @@ JAVA_TEST=Test.java
 CLASS_TEST=Test.class
 TEST=Test
 cat << \EOF > $JAVA_TEST
-/* [#]line 65167 "configure" */
+/* [#]line 65161 "configure" */
 public class Test {
 public static void main (String args[]) {
         System.exit (0);
@@ -67203,13 +67197,13 @@ HAS_WIDL_COMPILER_TRUE!$HAS_WIDL_COMPILE
 HAS_WIDL_COMPILER_FALSE!$HAS_WIDL_COMPILER_FALSE$ac_delim
 BUILD_ACTIVEX_TRUE!$BUILD_ACTIVEX_TRUE$ac_delim
 BUILD_ACTIVEX_FALSE!$BUILD_ACTIVEX_FALSE$ac_delim
-FIREFOX_CFLAGS!$FIREFOX_CFLAGS$ac_delim
-FIREFOX_LIBS!$FIREFOX_LIBS$ac_delim
+ICEAPE_CFLAGS!$ICEAPE_CFLAGS$ac_delim
+ICEAPE_LIBS!$ICEAPE_LIBS$ac_delim
 MOZILLA_CONFIG!$MOZILLA_CONFIG$ac_delim
 SEAMONKEY_CONFIG!$SEAMONKEY_CONFIG$ac_delim
 XULRUNNER_CONFIG!$XULRUNNER_CONFIG$ac_delim
-FIREFOX_XPCOM_CFLAGS!$FIREFOX_XPCOM_CFLAGS$ac_delim
-FIREFOX_XPCOM_LIBS!$FIREFOX_XPCOM_LIBS$ac_delim
+ICEAPE_XPCOM_CFLAGS!$ICEAPE_XPCOM_CFLAGS$ac_delim
+ICEAPE_XPCOM_LIBS!$ICEAPE_XPCOM_LIBS$ac_delim
 MOZILLA_NSPR_CFLAGS!$MOZILLA_NSPR_CFLAGS$ac_delim
 MOZILLA_NSPR_LIBS!$MOZILLA_NSPR_LIBS$ac_delim
 NSPR_CFLAGS!$NSPR_CFLAGS$ac_delim
