diff --git a/share/Makefile.in b/share/Makefile.in
index 84090b9..1395603 100644
--- a/share/Makefile.in
+++ b/share/Makefile.in
@@ -627,7 +636,7 @@ top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 vlcdatadir = @vlcdatadir@
 vlclibdir = @vlclibdir@
-MOSTLYCLEANFILES = $(nobase_vlcdata_DATA)
+MOSTLYCLEANFILES = $(nodist_EXTRA_DATA)
 nobase_dist_data_DATA = applications/vlc.desktop
 dist_vlcdata_DATA = \
 	$(DIST_icons) \
@@ -642,13 +651,15 @@ nobase_dist_vlcdata_SCRIPTS = \
 	utils/video-vlc-default.sh \
 	$(NULL)
 
-nobase_vlcdata_DATA = $(NULL) $(am__append_1) $(am__append_2) \
-	$(am__append_3) $(am__append_4) $(am__append_5) \
-	$(am__append_6) $(am__append_7) $(am__append_8)
 EXTRA_DIST = \
+	$(skins2_default_vlt_FILES)
+
+nobase_vlcdata_DATA = $(am__append_1) $(am__append_2) $(am__append_3) \
+	$(am__append_4) $(am__append_5) $(am__append_6) \
+	$(am__append_7) $(am__append_8)
+dist_EXTRA_DATA = \
 	$(DIST_rsrc) \
 	$(DIST_pda) \
-	$(skins2_default_vlt_FILES) \
 	$(DIST_skins2) \
 	$(DIST_http) \
 	$(DIST_http_lua) \
@@ -657,6 +668,9 @@ EXTRA_DIST = \
 	$(DIST_mozilla) \
 	$(DIST_maemo)
 
+nodist_EXTRA_DATA = \
+	 skins2/default.vlt
+
 DIST_rsrc = \
 	newres.h \
 	resource.h \
