diff --git a/modules/demux/live555.cpp b/modules/demux/live555.cpp
index 3c75243..90040d3 100644
--- a/modules/demux/live555.cpp
+++ b/modules/demux/live555.cpp
@@ -829,7 +829,8 @@ describe:
         else
         {
             const char *psz_tmp = strstr( psz_error, "RTSP" );
-            sscanf( psz_tmp, "RTSP/%*s%3u", &i_code );
+            if( psz_tmp )
+                sscanf( psz_tmp, "RTSP/%*u.%*u %3u", &i_code );
         }
         msg_Dbg( p_demux, "DESCRIBE failed with %d: [%s]", i_code, psz_error );
 
@@ -869,13 +870,6 @@ describe:
             p_sys->rtsp = NULL;
             goto createnew;
         }
-        else
-        {
-            msg_Dbg( p_demux, "connection timeout, retrying" );
-            if( p_sys->rtsp ) RTSPClient::close( p_sys->rtsp );
-            p_sys->rtsp = NULL;
-            goto createnew;
-        }
         i_ret = VLC_EGENERIC;
     }
 
