Author: Reinhard Tartler <siretart@tauware.de>
Subject: fix kfreebsd ftbfs in access file module

Found while testbuilding on asdfasdf.debian.net

diff --git a/modules/access/file.c b/modules/access/file.c
index 3531fa7..dfd435e 100644
--- a/modules/access/file.c
+++ b/modules/access/file.c
@@ -96,7 +96,7 @@ static bool IsRemote (int fd)
     if (fstatvfs (fd, &stf))
         return false;
     /* fstatvfs() is in POSIX, but MNT_LOCAL is not */
-    return !(s.f_flag & MNT_LOCAL);
+    return !(stf.f_flag & MNT_LOCAL);
 
 #elif defined (HAVE_LINUX_MAGIC_H)
     struct statfs stf;
