From 464c5784caf9462a021505775cdc94ee2404475d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?R=C3=A9mi=20Denis-Courmont?= <remi@remlab.net>
Date: Sat, 10 Apr 2010 16:53:36 +0300
Subject: [PATCH] Require xul >= 1.9.2 explicitly

---
 configure.ac |    9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

Index: vlc-1.0.5/configure.ac
===================================================================
--- vlc-1.0.5.orig/configure.ac
+++ vlc-1.0.5/configure.ac
@@ -5622,40 +5622,43 @@
     found=0
     if test -n "$MOZILLA_CFLAGS" -a -n "$MOZILLA_LIBS"
     then
       found=1
     else
       if test -n "$PKG_CONFIG"; then
 	for i in "${with_mozilla_pkg}" libxul {seamonkey,iceape,xulrunner,firefox,iceweasel,mozilla}-plugin; do
 	  echo "Trying to find $i package" >&AS_MESSAGE_LOG_FD
-	  if $PKG_CONFIG --exists --print-errors "$i" 2>&AS_MESSAGE_LOG_FD
+	  AC_MSG_CHECKING([for $i >= 1.9.2])
+	  if $PKG_CONFIG --exists --print-errors "$i >= 1.9.2" 2>&AS_MESSAGE_LOG_FD
 	  then
             echo "Using $i pkg-config package." >&AS_MESSAGE_LOG_FD
             echo "Using $i package." >&AS_MESSAGE_FD
 	    found=1
 	    MOZILLA_CFLAGS=$(eval $PKG_CONFIG --cflags "$i" )
 	    MOZILLA_LIBS=$(eval $PKG_CONFIG --libs "$i" )
+	    AC_MSG_RESULT([yes])
 	    break
 	  fi
+	  AC_MSG_RESULT([no])
 	done
       fi
     fi
     AS_IF( [test $found = 1],[
         CPPFLAGS="${CPPFLAGS_save} ${MOZILLA_CFLAGS}"
         MOZILLA_REQUIRED_HEADERS=1
         AC_CHECK_HEADERS([npfunctions.h])
         AC_CHECK_HEADERS(npapi.h,,MOZILLA_REQUIRED_HEADERS=0)
         AC_CHECK_HEADERS(npruntime.h,,MOZILLA_REQUIRED_HEADERS=0,
           [#if HAVE_NPAPI_H
            # include <npapi.h>
            #endif
           ])
         if test "${MOZILLA_REQUIRED_HEADERS}" = "0"; then
-            AC_MSG_ERROR([Please install the Firefox development tools; mozilla-config.h, plugin/npapi.h and plugin/npruntime.h were not found.])
+            AC_MSG_ERROR([Please install the Firefox development tools; plugin/npapi.h and/or plugin/npruntime.h were not found.])
         fi
         MOZILLA_REQUIRED_HEADERS=
         mozilla=:
         AS_IF([ test "${SYS}" != "mingw32" -a "${SYS}" != "mingwce"],[
 	   VLC_ADD_CPPFLAGS([mozilla],[-DXP_UNIX -DOJI])
 	   AS_IF([ test "${SYS}" != "darwin"],[
 	     PKG_CHECK_MODULES(XPM, [xpm xt],[
 		VLC_ADD_CPPFLAGS([mozilla],[-DMOZ_X11])
@@ -5670,17 +5673,17 @@
         PLUGINS_BINDINGS="${PLUGINS_BINDINGS} mozilla"
         MOZILLA_CONFIG=
         CPPFLAGS="${CPPFLAGS_save}"
           ],
           [
             AC_PATH_PROGS(MOZILLA_CONFIG,
 	    [mozilla-config seamonkey-config xulrunner-config],
         [no])
-            test "${MOZILLA_CONFIG}" = "no" && AC_MSG_ERROR([Please install the Mozilla development tools. mozilla-config was not found.])
+            test "${MOZILLA_CONFIG}" = "no" && AC_MSG_ERROR([Please install the Mozilla development tools (version 1.9.2 or higher) or use --disable-mozilla.])
           ]
         )
     dnl pkg-config failed but we might have found a mozilla-config
     AS_IF( [test ! -z "${MOZILLA_CONFIG}"],[
       if ${MOZILLA_CONFIG} --defines | grep -q 'MOZ_X11=1'; then
         LDFLAGS="${LDFLAGS_save} ${X_LIBS} ${X_PRE_LIBS}"
         AC_CHECK_LIB(Xt,XtStrings,
          [
