From: siretart@tauware.de
Subject: Weaken dependencies on libschroedinger
 This allows vlc to build on debian/squeeze.
Forwarded: not-needed

diff --git a/configure.ac b/configure.ac
index f961846..a958d20 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3068,7 +3068,7 @@ PKG_ENABLE_MODULES_VLC([DIRAC], [], [dirac >= 0.10.0], [dirac encoder], [auto])
 dnl
 dnl  schroedinger decoder plugin (for dirac format video)
 dnl
-PKG_ENABLE_MODULES_VLC([SCHROEDINGER], [], [schroedinger-1.0 >= 1.0.10], [dirac decoder and encoder using schroedinger], [auto])
+PKG_ENABLE_MODULES_VLC([SCHROEDINGER], [], [schroedinger-1.0 >= 1.0.9], [dirac decoder and encoder using schroedinger], [auto])
 
 dnl
 dnl  PNG decoder module
