--- a/modules/hw/mmal/codec.c
+++ b/modules/hw/mmal/codec.c
@@ -29,6 +29,7 @@
 #include <stdatomic.h>
 
 #include <vlc_common.h>
+#include <vlc_cpu.h>
 #include <vlc_plugin.h>
 #include <vlc_codec.h>
 #include <vlc_filter.h>
@@ -1869,6 +1870,9 @@
                 p_filter->fmt_out.video.i_visible_width, p_filter->fmt_out.video.i_visible_height);
     }
 
+    if (!vlc_CPU_ARM_NEON())
+        return VLC_EGENERIC;
+
     {
         blend_sys_t * const sys = calloc(1, sizeof (*sys));
         if (sys == NULL)
