From: Rémi Denis-Courmont <remi@remlab.net>
Subject: ASX: fix NULL derefence
Applied-Upstream: 1.1.10
Origin: upstream, http://git.videolan.org/?p=vlc%2Fvlc-1.1.git;h=6849c233ac434403909da29e0861875abebcc718
Bug-Ubuntu: https://launchpad.net/bugs/785979

--- a/modules/demux/playlist/asx.c
+++ b/modules/demux/playlist/asx.c
@@ -439,11 +439,15 @@
                     }
                     else continue;
                 }
-                if( ( psz_parse = strcasestr( psz_parse, "/>" ) ) )
-                    psz_parse += 2;
-                else if( ( psz_parse = strcasestr( psz_parse, "</MoreInfo>") ) )
-                    psz_parse += 11;
-                else continue;
+                if( ( psz_backup = strcasestr( psz_parse, "/>" ) ) )
+                    psz_parse = psz_backup + 2;
+                else if( ( psz_backup = strcasestr( psz_parse, "</MoreInfo>") ) )
+                    psz_parse = psz_backup + 11;
+                else
+                {
+                    psz_parse = NULL;
+                    continue;
+                }
             }
             else if( !strncasecmp( psz_parse, "<ABSTRACT>", 10 ) )
             {
@@ -483,7 +487,6 @@
                 }
                 if( ( psz_parse = strcasestr( psz_parse, "/>" ) ) )
                     psz_parse += 2;
-                else continue;
             }
             else if( !strncasecmp( psz_parse, "</Entry>", 8 ) )
             {
