From: Pierre Ynard <linkfanel@yahoo.fr>
Subject: lirc: build fix
Origin: upstream, http://git.videolan.org/?p=vlc/vlc-1.1.git;a=commitdiff;h=b5569228dd32aa848d2c8eed7e3f50d3a8522b8d
Applied-Upstream: 1.1.7

diff --git a/modules/control/lirc.c b/modules/control/lirc.c
index aee33c3..e8c4780 100644
--- a/modules/control/lirc.c
+++ b/modules/control/lirc.c
@@ -25,6 +25,7 @@
  * Preamble
  *****************************************************************************/
 
+#include <errno.h>
 #include <fcntl.h>
 
 #ifdef HAVE_CONFIG_H
@@ -154,10 +155,12 @@ static void Run( intf_thread_t *p_intf )
         /* Wait for data */
         struct pollfd ufd = { .fd = p_sys->i_fd, .events = POLLIN, .revents = 0 };
         if( poll( &ufd, 1, -1 ) == -1 )
+        {
             if( errno == EINTR )
                 continue;
             else
                 break;
+        }
 
         /* Process */
         int canc = vlc_savecancel();
