From: Dominique Leuenberger <dominique@leuenberger.net>
Subject: Libnotify depends on gtk, but leaves the choice between gtk2 and gtk3
 to the calling...
Origin: backport, http://git.videolan.org/?p=vlc.git;a=commitdiff;h=f62e6d6e7ca732471f84fc766988553974b8a1b8
Applied-Upstream: 1.2.0

--- a/configure.ac
+++ b/configure.ac
@@ -1721,7 +1721,7 @@
 dnl
 dnl Libnotify notification plugin
 dnl
-PKG_ENABLE_MODULES_VLC([NOTIFY], [], [libnotify], [libnotify notification], [auto])
+PKG_ENABLE_MODULES_VLC([NOTIFY], [], [libnotify gtk+-2.0], [libnotify notification], [auto])
 
 dnl
 dnl Taglibplugin
