diff -puriN vlc-0.8.6/modules/codec/sdl_image.c vlc-0.8.6a/modules/codec/sdl_image.c
--- vlc-0.8.6/modules/codec/sdl_image.c	2006-09-18 12:28:28.000000000 +0200
+++ vlc-0.8.6a/modules/codec/sdl_image.c	2007-01-07 23:07:35.000000000 +0100
@@ -52,7 +52,7 @@ vlc_module_begin();
     set_category( CAT_INPUT );
     set_subcategory( SUBCAT_INPUT_VCODEC );
     set_description( _("SDL_image video decoder") );
-    set_capability( "decoder", 900 );
+    set_capability( "decoder", 60 );
     set_callbacks( OpenDecoder, CloseDecoder );
     add_shortcut( "sdl_image" );
 vlc_module_end();
