diff -puriN vlc-0.8.6/src/playlist/item-ext.c vlc-0.8.6a/src/playlist/item-ext.c
--- vlc-0.8.6/src/playlist/item-ext.c	2006-09-18 12:27:56.000000000 +0200
+++ vlc-0.8.6a/src/playlist/item-ext.c	2007-01-07 23:06:39.000000000 +0100
@@ -734,7 +734,7 @@ int playlist_Clear( playlist_t * p_playl
     int i;
     for( i = p_playlist->i_size; i > 0 ; i-- )
     {
-        playlist_Delete( p_playlist, p_playlist->pp_items[0]->input.i_id );
+        playlist_LockDelete( p_playlist, p_playlist->pp_items[0]->input.i_id );
     }
     for( i = 0 ; i< p_playlist->i_views; i++ )
     {
diff -puriN vlc-0.8.6/src/playlist/playlist.c vlc-0.8.6a/src/playlist/playlist.c
--- vlc-0.8.6/src/playlist/playlist.c	2006-09-18 12:27:56.000000000 +0200
+++ vlc-0.8.6a/src/playlist/playlist.c	2007-01-07 23:06:39.000000000 +0100
@@ -657,6 +657,8 @@ static void RunThread ( playlist_t *p_pl
                 /* Destroy input */
                 input_DestroyThread( p_input );
 
+                vlc_mutex_lock( &p_playlist->object_lock );
+
                 /* Unlink current input
                  * (_after_ input_DestroyThread for vout garbage collector) */
                 vlc_object_detach( p_input );
@@ -674,6 +676,7 @@ static void RunThread ( playlist_t *p_pl
                      p_playlist->status.p_item = NULL;
                 }
 
+                vlc_mutex_unlock( &p_playlist->object_lock );
                 continue;
             }
             /* This input is dying, let it do */
@@ -796,12 +799,16 @@ static void RunThread ( playlist_t *p_pl
 
             /* Destroy input */
             input_DestroyThread( p_input );
+
+            vlc_mutex_lock( &p_playlist->object_lock );
+
             /* Unlink current input (_after_ input_DestroyThread for vout
              * garbage collector)*/
             vlc_object_detach( p_input );
 
             /* Destroy object */
             vlc_object_destroy( p_input );
+            vlc_mutex_unlock( &p_playlist->object_lock );
             continue;
         }
         else if( p_playlist->p_input->b_die )
