diff -puriN vlc-0.8.6b/modules/access/cdda.c vlc-0.8.6c/modules/access/cdda.c
--- vlc-0.8.6b/modules/access/cdda.c	2007-04-17 23:45:05 +0200
+++ vlc-0.8.6c/modules/access/cdda.c	2007-06-16 16:25:11 +0200
@@ -630,18 +630,17 @@ static int GetTracks( access_t *p_access
                     {
                         vlc_input_item_AddInfo( &p_item->input,
                                             _(VLC_META_INFO_CAT),
-                                            _(VLC_META_TITLE),
+                                            _(VLC_META_TITLE), "%s", 
                                             cddb_track_get_title( t ) );
                         if( p_item->input.psz_name )
                             free( p_item->input.psz_name );
-                        asprintf( &p_item->input.psz_name, "%s",
-                                  cddb_track_get_title( t ) );
+                        p_item->input.psz_name = strdup( cddb_track_get_title( t ) );
                     }
                     psz_result = cddb_track_get_artist( t );
                     if( psz_result )
                     {
                         vlc_input_item_AddInfo( &p_item->input,
-                                            _(VLC_META_INFO_CAT),
+                                            _(VLC_META_INFO_CAT), "%s",
                                             _(VLC_META_ARTIST), psz_result );
                     }
                 }
diff -puriN vlc-0.8.6b/modules/codec/theora.c vlc-0.8.6c/modules/codec/theora.c
--- vlc-0.8.6b/modules/codec/theora.c	2007-04-17 23:45:11 +0200
+++ vlc-0.8.6c/modules/codec/theora.c	2007-06-16 16:25:12 +0200
@@ -510,7 +510,7 @@ static void ParseTheoraComments( decoder
             *psz_value = '\0';
             psz_value++;
             input_Control( p_input, INPUT_ADD_INFO, _("Theora comment"),
-                           psz_name, psz_value );
+                           psz_name, "%s", psz_value );
         }
         free( psz_comment );
         i++;
diff -puriN vlc-0.8.6b/modules/codec/vorbis.c vlc-0.8.6c/modules/codec/vorbis.c
--- vlc-0.8.6b/modules/codec/vorbis.c	2007-04-17 23:45:11 +0200
+++ vlc-0.8.6c/modules/codec/vorbis.c	2007-06-16 16:25:12 +0200
@@ -623,7 +623,7 @@ static void ParseVorbisComments( decoder
             *psz_value = '\0';
             psz_value++;
             input_Control( p_input, INPUT_ADD_INFO, _("Vorbis comment"),
-                           psz_name, psz_value );
+                           psz_name, "%s", psz_value );
             if( strcasestr( psz_name, "artist" ) )
             {
                 vlc_input_item_AddInfo( p_input->input.p_item,
diff -puriN vlc-0.8.6b/modules/services_discovery/sap.c vlc-0.8.6c/modules/services_discovery/sap.c
--- vlc-0.8.6b/modules/services_discovery/sap.c	2007-04-17 23:45:09 +0200
+++ vlc-0.8.6c/modules/services_discovery/sap.c	2007-06-16 16:25:12 +0200
@@ -840,12 +840,12 @@ sap_announce_t *CreateAnnounce( services
     if( psz_value != NULL )
     {
         vlc_input_item_AddInfo( &p_item->input, _("Session"),
-                                _("Tool"), psz_value );
+                                _("Tool"), "%s", psz_value );
     }
     if( strcmp( p_sdp->psz_username, "-" ) )
     {
         vlc_input_item_AddInfo( &p_item->input, _("Session"),
-                                _("User"), p_sdp->psz_username );
+                                _("User"), "%s", p_sdp->psz_username );
     }
 
     psz_value = GetAttribute( p_sap->p_sdp, "x-plgroup" );
