diff -puriN vlc-0.8.6/modules/video_filter/logo.c vlc-0.8.6a/modules/video_filter/logo.c
--- vlc-0.8.6/modules/video_filter/logo.c	2006-10-12 19:47:07.000000000 +0200
+++ vlc-0.8.6a/modules/video_filter/logo.c	2007-01-07 23:07:30.000000000 +0100
@@ -265,11 +265,11 @@ void FreeLogoList( logo_list_t *p_logo_l
     for( i = 0; i < p_logo_list->i_count; i++ )
     {
         logo_t *p_logo = &p_logo_list->p_logo[i];
-        if( p_logo[i].psz_file ) FREE( p_logo[i].psz_file );
-        if( p_logo[i].p_pic )
+        if( p_logo->psz_file ) FREE( p_logo->psz_file );
+        if( p_logo->p_pic )
         {
-            p_logo[i].p_pic->pf_release( p_logo[i].p_pic );
-            p_logo[i].p_pic = NULL;
+            p_logo->p_pic->pf_release( p_logo->p_pic );
+            p_logo->p_pic = NULL;
         }
     }
 }
