Index: vlc-0.8.6-svn20061012.debian/modules/access/dv.c
===================================================================
--- vlc-0.8.6-svn20061012.debian.orig/modules/access/dv.c	2006-09-18 12:28:26.000000000 +0200
+++ vlc-0.8.6-svn20061012.debian/modules/access/dv.c	2007-01-08 02:29:13.000000000 +0100
@@ -303,14 +303,14 @@
     switch( i_query )
     {
         /* */
-        case ACCESS_CAN_SEEK:
-        case ACCESS_CAN_FASTSEEK:
         case ACCESS_CAN_PAUSE:
             pb_bool = (vlc_bool_t*)va_arg( args, vlc_bool_t* );
             *pb_bool = VLC_TRUE;
             break;
 
-        case ACCESS_CAN_CONTROL_PACE:
+       case ACCESS_CAN_SEEK:
+       case ACCESS_CAN_FASTSEEK:
+       case ACCESS_CAN_CONTROL_PACE:
             pb_bool = (vlc_bool_t*)va_arg( args, vlc_bool_t* );
             *pb_bool = VLC_FALSE;
             break;
Index: vlc-0.8.6-svn20061012.debian/modules/access/vcdx/access.c
===================================================================
--- vlc-0.8.6-svn20061012.debian.orig/modules/access/vcdx/access.c	2007-01-08 02:28:25.000000000 +0100
+++ vlc-0.8.6-svn20061012.debian/modules/access/vcdx/access.c	2007-01-08 02:29:13.000000000 +0100
@@ -1219,5 +1219,6 @@
             return VLC_EGENERIC;
 
     }
+    p_access->info.b_eof = VLC_FALSE;
     return VLC_SUCCESS;
 }
Index: vlc-0.8.6-svn20061012.debian/modules/audio_filter/resampler/linear.c
===================================================================
--- vlc-0.8.6-svn20061012.debian.orig/modules/audio_filter/resampler/linear.c	2006-09-18 12:28:27.000000000 +0200
+++ vlc-0.8.6-svn20061012.debian/modules/audio_filter/resampler/linear.c	2007-01-08 02:29:13.000000000 +0100
@@ -102,7 +102,7 @@
         return VLC_ENOMEM;
     }
     p_sys->p_prev_sample = malloc(
-        aout_FormatNbChannels( &p_filter->input ) * sizeof(int32_t) );
+        p_filter->input.i_channels * sizeof(int32_t) );
     if( p_sys->p_prev_sample == NULL )
     {
         msg_Err( p_filter, "out of memory" );
@@ -143,7 +143,7 @@
     float *p_in, *p_out = (float *)p_out_buf->p_buffer;
     float *p_prev_sample = (float *)p_sys->p_prev_sample;
 
-    int i_nb_channels = aout_FormatNbChannels( &p_filter->input );
+    int i_nb_channels = p_filter->input.i_channels;
     int i_in_nb = p_in_buf->i_nb_samples;
     int i_chan, i_in, i_out = 0;
 
Index: vlc-0.8.6-svn20061012.debian/modules/codec/flac.c
===================================================================
--- vlc-0.8.6-svn20061012.debian.orig/modules/codec/flac.c	2006-10-08 18:40:57.000000000 +0200
+++ vlc-0.8.6-svn20061012.debian/modules/codec/flac.c	2007-01-08 02:29:13.000000000 +0100
@@ -97,7 +97,7 @@
     STATE_SEND_DATA
 };
 
-static int pi_channels_maps[6] =
+static int pi_channels_maps[7] =
 {
     0,
     AOUT_CHAN_CENTER,
@@ -106,7 +106,9 @@
     AOUT_CHAN_LEFT | AOUT_CHAN_RIGHT | AOUT_CHAN_REARLEFT
      | AOUT_CHAN_REARRIGHT,
     AOUT_CHAN_LEFT | AOUT_CHAN_RIGHT | AOUT_CHAN_CENTER
-     | AOUT_CHAN_REARLEFT | AOUT_CHAN_REARRIGHT
+     | AOUT_CHAN_REARLEFT | AOUT_CHAN_REARRIGHT,
+    AOUT_CHAN_LEFT | AOUT_CHAN_RIGHT | AOUT_CHAN_CENTER
+     | AOUT_CHAN_REARLEFT | AOUT_CHAN_REARRIGHT | AOUT_CHAN_LFE
 };
 
 /*****************************************************************************
@@ -285,7 +287,7 @@
 }
 
 /*****************************************************************************
- * ProcessHeader: processe Flac header.
+ * ProcessHeader: process Flac header.
  *****************************************************************************/
 static void ProcessHeader( decoder_t *p_dec )
 {
@@ -347,6 +349,12 @@
 
     if( !p_sys->b_stream_info ) ProcessHeader( p_dec );
 
+    if( p_sys->stream_info.channels > 6 )
+    {
+        msg_Err( p_dec, "This stream uses too many audio channels" );
+        return NULL;
+    }
+
     if( !aout_DateGet( &p_sys->end_date ) && !(*pp_block)->i_pts )
     {
         /* We've just started the stream, wait for the first PTS. */
Index: vlc-0.8.6-svn20061012.debian/modules/codec/libmpeg2.c
===================================================================
--- vlc-0.8.6-svn20061012.debian.orig/modules/codec/libmpeg2.c	2006-09-18 12:28:28.000000000 +0200
+++ vlc-0.8.6-svn20061012.debian/modules/codec/libmpeg2.c	2007-01-08 02:29:13.000000000 +0100
@@ -64,8 +64,9 @@
     vlc_bool_t       b_after_sequence_header; /* is it the next frame after
                                                * the sequence header ?    */
     vlc_bool_t       b_slice_i;             /* intra-slice refresh stream */
+    vlc_bool_t       b_second_field;
 
-    vlc_bool_t      b_preroll;
+    vlc_bool_t       b_preroll;
 
     /*
      * Output properties
@@ -116,6 +117,7 @@
         p_dec->fmt_in.i_codec != VLC_FOURCC('P','I','M','1') &&
         /* ATI Video */
         p_dec->fmt_in.i_codec != VLC_FOURCC('V','C','R','2') &&
+        p_dec->fmt_in.i_codec != VLC_FOURCC('m','p','2','v') &&
         p_dec->fmt_in.i_codec != VLC_FOURCC('m','p','g','2') &&
         p_dec->fmt_in.i_codec != VLC_FOURCC('h','d','v','2') )
     {
@@ -142,6 +144,7 @@
     p_sys->p_picture_to_destroy = NULL;
     p_sys->b_garbage_pic = 0;
     p_sys->b_slice_i  = 0;
+    p_sys->b_second_field = 0;
     p_sys->b_skip     = 0;
     p_sys->b_preroll = VLC_FALSE;
 
@@ -338,20 +341,7 @@
         break;
 
         case STATE_PICTURE_2ND:
-            vout_SynchroNewPicture( p_sys->p_synchro,
-                p_sys->p_info->current_picture->flags & PIC_MASK_CODING_TYPE,
-                p_sys->p_info->current_picture->nb_fields,
-                0, 0, p_sys->i_current_rate,
-                p_sys->p_info->sequence->flags & SEQ_FLAG_LOW_DELAY );
-
-            if( p_sys->b_skip )
-            {
-                vout_SynchroTrash( p_sys->p_synchro );
-            }
-            else
-            {
-                vout_SynchroDecode( p_sys->p_synchro );
-            }
+            p_sys->b_second_field = 1;
             break;
 
         case STATE_PICTURE:
@@ -407,8 +397,15 @@
                   p_sys->i_current_dts : p_sys->i_previous_dts ) : 0;
 #endif
 
+            /* If nb_fields == 1, it is a field picture, and it will be
+             * followed by another field picture for which we won't call
+             * vout_SynchroNewPicture() because this would have other 
+             * problems, so we take it into account here.
+             * This kind of sucks, but I didn't think better. --Meuuh
+             */
             vout_SynchroNewPicture( p_sys->p_synchro,
                 p_sys->p_info->current_picture->flags & PIC_MASK_CODING_TYPE,
+                p_sys->p_info->current_picture->nb_fields == 1 ? 2 :
                 p_sys->p_info->current_picture->nb_fields, i_pts, i_dts,
                 p_sys->i_current_rate,
                 p_sys->p_info->sequence->flags & SEQ_FLAG_LOW_DELAY );
Index: vlc-0.8.6-svn20061012.debian/modules/codec/realaudio.c
===================================================================
--- vlc-0.8.6-svn20061012.debian.orig/modules/codec/realaudio.c	2006-09-18 12:28:28.000000000 +0200
+++ vlc-0.8.6-svn20061012.debian/modules/codec/realaudio.c	2007-01-08 02:29:13.000000000 +0100
@@ -185,6 +185,14 @@
         return VLC_EGENERIC;
     }
 
+    if( p_dec->fmt_in.audio.i_channels <= 0 ||
+        p_dec->fmt_in.audio.i_channels > 6 )
+    {
+        msg_Err( p_dec, "invalid number of channels (not between 1 and 6): %i",
+                 p_dec->fmt_in.audio.i_channels );
+        return VLC_EGENERIC;
+    }
+
     p_dec->p_sys = p_sys = malloc( sizeof( decoder_sys_t ) );
     memset( p_sys, 0, sizeof(decoder_sys_t) );
 
Index: vlc-0.8.6-svn20061012.debian/modules/codec/speex.c
===================================================================
--- vlc-0.8.6-svn20061012.debian.orig/modules/codec/speex.c	2006-09-18 12:28:28.000000000 +0200
+++ vlc-0.8.6-svn20061012.debian/modules/codec/speex.c	2007-01-08 02:29:13.000000000 +0100
@@ -383,6 +383,13 @@
         callback.data = &p_sys->stereo;
         speex_decoder_ctl( p_state, SPEEX_SET_HANDLER, &callback );
     }
+    if( p_header->nb_channels <= 0 ||
+        p_header->nb_channels > 5 )
+    {
+        msg_Err( p_dec, "invalid number of channels (not between 1 and 5): %i",
+                 p_header->nb_channels );
+        return VLC_EGENERIC;
+    }
 
     /* Setup the format */
     p_dec->fmt_out.audio.i_physical_channels =
Index: vlc-0.8.6-svn20061012.debian/modules/codec/subsdec.c
===================================================================
--- vlc-0.8.6-svn20061012.debian.orig/modules/codec/subsdec.c	2006-09-22 17:14:32.000000000 +0200
+++ vlc-0.8.6-svn20061012.debian/modules/codec/subsdec.c	2007-01-08 02:29:13.000000000 +0100
@@ -316,9 +316,11 @@
     }
 
     /* Check validity of packet data */
-    if( p_block->i_buffer <= 1 || p_block->p_buffer[0] == '\0' )
+    /* An "empty" line containing only \0 can be used to force
+       and ephemer picture from the screen */
+    if( p_block->i_buffer < 1 )
     {
-        msg_Warn( p_dec, "empty subtitle" );
+        msg_Warn( p_dec, "no subtitle data" );
         return NULL;
     }
 
Index: vlc-0.8.6-svn20061012.debian/modules/codec/vorbis.c
===================================================================
--- vlc-0.8.6-svn20061012.debian.orig/modules/codec/vorbis.c	2006-09-18 12:28:28.000000000 +0200
+++ vlc-0.8.6-svn20061012.debian/modules/codec/vorbis.c	2007-01-08 02:29:13.000000000 +0100
@@ -387,6 +387,15 @@
     /* Setup the format */
     p_dec->fmt_out.audio.i_rate     = p_sys->vi.rate;
     p_dec->fmt_out.audio.i_channels = p_sys->vi.channels;
+
+    if( p_dec->fmt_out.audio.i_channels < 0 ||
+        p_dec->fmt_out.audio.i_channels > 6 )
+    {
+        msg_Err( p_dec, "invalid number of channels (not between 1 and 6): %i",
+                 p_dec->fmt_out.audio.i_channels );
+        return VLC_EGENERIC;
+    }
+
     p_dec->fmt_out.audio.i_physical_channels =
         p_dec->fmt_out.audio.i_original_channels =
             pi_channels_maps[p_sys->vi.channels];
Index: vlc-0.8.6-svn20061012.debian/modules/demux/avi/avi.c
===================================================================
--- vlc-0.8.6-svn20061012.debian.orig/modules/demux/avi/avi.c	2006-10-08 18:40:53.000000000 +0200
+++ vlc-0.8.6-svn20061012.debian/modules/demux/avi/avi.c	2007-01-08 02:29:13.000000000 +0100
@@ -53,7 +53,7 @@
 
 static int pi_index[] = {0,1,2};
 
-static char *ppsz_indexes[] = { N_("Ask"),N_("Always fix"),
+static const char *ppsz_indexes[] = { N_("Ask"), N_("Always fix"),
                                 N_("Never fix") };
 
 vlc_module_begin();
@@ -798,7 +798,7 @@
             }
             if( toread[i].i_posf > 0 )
             {
-                if( i_pos == -1 || i_pos > toread[i_track].i_posf )
+                if( i_pos == -1 || i_pos > toread[i].i_posf )
                 {
                     i_track = i;
                     i_pos = toread[i].i_posf;
@@ -1305,7 +1305,7 @@
                 }
             }
         }
-        return (double)i64 / (double)stream_Size( p_demux->s );
+        return (double)i64 / stream_Size( p_demux->s );
     }
     return 0.0;
 }
Index: vlc-0.8.6-svn20061012.debian/modules/demux/ps.c
===================================================================
--- vlc-0.8.6-svn20061012.debian.orig/modules/demux/ps.c	2006-09-21 17:50:22.000000000 +0200
+++ vlc-0.8.6-svn20061012.debian/modules/demux/ps.c	2007-01-08 02:29:13.000000000 +0100
@@ -56,7 +56,7 @@
     set_callbacks( Open, Close );
     add_shortcut( "ps" );
 
-    add_integer( "ps-trust-timestamps", VLC_TRUE, NULL, TIME_TEXT,
+    add_bool( "ps-trust-timestamps", VLC_TRUE, NULL, TIME_TEXT,
                  TIME_LONGTEXT, VLC_TRUE );
 
     add_submodule();
@@ -184,7 +184,7 @@
     free( p_sys );
 }
 
-static int Demux2( demux_t *p_demux )
+static int Demux2( demux_t *p_demux, vlc_bool_t b_end )
 {
     demux_sys_t *p_sys = p_demux->p_sys;
     int i_ret, i_id;
@@ -217,7 +217,14 @@
         ps_track_t *tk = &p_sys->tk[PS_ID_TO_TK(i_id)];
         if( !ps_pkt_parse_pes( p_pkt, tk->i_skip ) )
         {
-            tk->i_last_pts = p_pkt->i_pts;
+            if( b_end && p_pkt->i_pts > tk->i_last_pts )
+            {
+                tk->i_last_pts = p_pkt->i_pts;
+            }
+            else if ( tk->i_first_pts == -1 )
+            {
+                tk->i_first_pts = p_pkt->i_pts;
+            }
         }
     }
     block_Release( p_pkt );
@@ -235,18 +242,25 @@
 
     if( p_sys->i_length == -1 ) /* First time */
     {
+        p_sys->i_length = 0;
+        /* Check beginning */
+        i = 0;
         i_current_pos = stream_Tell( p_demux->s );
+        while( !p_demux->b_die && i < 40 && Demux2( p_demux, VLC_FALSE ) > 0 ) i++;
+
+        /* Check end */
         i_size = stream_Size( p_demux->s );
-        i_end = __MAX( 0, __MIN( 100000, i_size ) );
-        stream_Seek( p_demux->s, stream_Size( p_demux->s ) - i_end );
+        i_end = __MAX( 0, __MIN( 200000, i_size ) );
+        stream_Seek( p_demux->s, i_size - i_end );
     
-        while( Demux2( p_demux ) > 0 && !p_demux->b_die ) ;
+        while( !p_demux->b_die && Demux2( p_demux, VLC_TRUE ) > 0 );
+        if( i_current_pos >= 0 ) stream_Seek( p_demux->s, i_current_pos );
     }
 
     for( i = 0; i < PS_TK_COUNT; i++ )
     {
         ps_track_t *tk = &p_sys->tk[i];
-        if( tk->b_seen && tk->i_first_pts > 0 && tk->i_last_pts > 0 )
+        if( tk->i_first_pts >= 0 && tk->i_last_pts > 0 )
             if( tk->i_last_pts > tk->i_first_pts )
             {
                 int64_t i_length = (int64_t)tk->i_last_pts - tk->i_first_pts;
@@ -254,10 +268,10 @@
                 {
                     p_sys->i_length = i_length;
                     p_sys->i_time_track = i;
+                    msg_Dbg( p_demux, "we found a length of: %lld", p_sys->i_length );
                 }
             }
     }
-    if( i_current_pos >= 0 ) stream_Seek( p_demux->s, i_current_pos );
 }
 
 /*****************************************************************************
@@ -287,7 +301,7 @@
     if( p_sys->b_lost_sync ) msg_Warn( p_demux, "found sync code" );
     p_sys->b_lost_sync = VLC_FALSE;
 
-    if( p_sys->b_seekable && p_sys->i_length <= 0 )
+    if( p_sys->i_length < 0 && p_sys->b_seekable )
         FindLength( p_demux );
 
     if( ( p_pkt = ps_pkt_read( p_demux->s, i_code ) ) == NULL )
@@ -381,11 +395,6 @@
                     es_out_Control( p_demux->out, ES_OUT_SET_PCR, p_pkt->i_pts );
                 }
 
-                if( !tk->i_first_pts && p_pkt->i_pts > 0 )
-                {
-                    tk->i_first_pts = p_pkt->i_pts;
-                    p_sys->i_length = 0;
-                }
                 if( (int64_t)p_pkt->i_pts > p_sys->i_current_pts )
                 {
                     p_sys->i_current_pts = (int64_t)p_pkt->i_pts;
@@ -473,6 +482,19 @@
             return VLC_EGENERIC;
 
         case DEMUX_SET_TIME:
+            i64 = (int64_t)va_arg( args, int64_t );
+            if( p_sys->i_time_track >= 0 && p_sys->i_current_pts > 0 )
+            {
+                int64_t i_now = p_sys->i_current_pts - p_sys->tk[p_sys->i_time_track].i_first_pts;
+                int64_t i_pos = stream_Tell( p_demux->s );
+                int64_t i_offset = i_pos / (i_now / 1000000) * ((i64 - i_now) / 1000000);
+                stream_Seek( p_demux->s, i_pos + i_offset);
+
+                es_out_Control( p_demux->out, ES_OUT_RESET_PCR );
+                return VLC_SUCCESS;
+            }
+            return VLC_EGENERIC;
+
         case DEMUX_GET_FPS:
         default:
             return VLC_EGENERIC;
Index: vlc-0.8.6-svn20061012.debian/modules/demux/ps.h
===================================================================
--- vlc-0.8.6-svn20061012.debian.orig/modules/demux/ps.h	2006-09-18 12:28:17.000000000 +0200
+++ vlc-0.8.6-svn20061012.debian/modules/demux/ps.h	2007-01-08 02:29:13.000000000 +0100
@@ -51,8 +51,8 @@
         tk[i].i_skip = 0;
         tk[i].i_id   = 0;
         tk[i].es     = NULL;
-        tk[i].i_first_pts = 0;
-        tk[i].i_last_pts = 0;
+        tk[i].i_first_pts = -1;
+        tk[i].i_last_pts = -1;
         es_format_Init( &tk[i].fmt, UNKNOWN_ES, 0 );
     }
 }
Index: vlc-0.8.6-svn20061012.debian/modules/demux/subtitle.c
===================================================================
--- vlc-0.8.6-svn20061012.debian.orig/modules/demux/subtitle.c	2006-10-12 19:47:03.000000000 +0200
+++ vlc-0.8.6-svn20061012.debian/modules/demux/subtitle.c	2007-01-08 02:29:13.000000000 +0100
@@ -201,6 +201,7 @@
     if( f_fps >= 1.0 )
     {
         p_sys->i_microsecperframe = (int64_t)( (float)1000000 / f_fps );
+        msg_Dbg( p_demux, "Override subtitle fps %f", f_fps );
     }
 
     p_input = (input_thread_t *)vlc_object_find( p_demux, VLC_OBJECT_INPUT, FIND_PARENT );
@@ -210,6 +211,7 @@
         if( f_fps >= 1.0 )
             p_sys->i_original_mspf = (int64_t)( (float)1000000 / f_fps );
 
+        msg_Dbg( p_demux, "Movie fps: %f", f_fps );
         vlc_object_release( p_input );
     }
 
@@ -701,7 +703,8 @@
     int    i_stop;
     unsigned int i;
 
-    int i_microsecperframe = 40000; /* default to 25 fps */
+    /* Try sub-fps value if set, movie rate if know, else 25fps (40000) */
+    int i_microsecperframe = p_sys->i_original_mspf > 0 ? p_sys->i_original_mspf : 40000;
     if( p_sys->i_microsecperframe > 0 )
         i_microsecperframe = p_sys->i_microsecperframe;
 
@@ -729,8 +732,9 @@
     if( i_start == 1 && i_stop == 1 )
     {
         /* We found a possible setting of the framerate "{1}{1}23.976" */
+        /* Use it unless sub-fps was set */
         float tmp = us_strtod( buffer_text, NULL );
-        if( tmp > 0.0 && !var_GetFloat( p_demux, "sub-fps" ) > 0.0 )
+        if( tmp > 0.0 && var_GetFloat( p_demux, "sub-fps" ) <= 0.0 )
             p_sys->i_microsecperframe = (int64_t)( (float)1000000 / tmp );
         goto next;
     }
Index: vlc-0.8.6-svn20061012.debian/modules/demux/ts.c
===================================================================
--- vlc-0.8.6-svn20061012.debian.orig/modules/demux/ts.c	2006-09-18 12:28:17.000000000 +0200
+++ vlc-0.8.6-svn20061012.debian/modules/demux/ts.c	2007-01-08 02:29:13.000000000 +0100
@@ -2647,6 +2647,8 @@
     msg_Dbg( p_demux, "PSINewTableCallBack: table 0x%x(%d) ext=0x%x(%d)",
              i_table_id, i_table_id, i_extension, i_extension );
 #endif
+    if( p_demux->p_sys->pid[0].psi->i_pat_version == -1 )
+	return;
 
     if( i_table_id == 0x42 )
     {
@@ -3270,7 +3272,8 @@
             }
         }
 
-        if( DVBProgramIsSelected( p_demux, prg->i_number ) )
+        if( DVBProgramIsSelected( p_demux, prg->i_number )
+             && (pid->es->id != NULL || p_sys->b_udp_out) )
         {
             /* Set demux filter */
             stream_Control( p_demux->s, STREAM_CONTROL_ACCESS,
@@ -3467,7 +3470,9 @@
                         if( p_sys->i_dvb_program == 0 )
                             p_sys->i_dvb_program = p_program->i_number;
 
-                        if( stream_Control( p_demux->s, STREAM_CONTROL_ACCESS, ACCESS_SET_PRIVATE_ID_STATE, p_program->i_pid, VLC_TRUE ) )
+                        if( stream_Control( p_demux->s, STREAM_CONTROL_ACCESS,
+                                            ACCESS_SET_PRIVATE_ID_STATE,
+                                            p_program->i_pid, VLC_TRUE ) )
                             p_sys->b_dvb_control = VLC_FALSE;
                     }
                 }
Index: vlc-0.8.6-svn20061012.debian/modules/mux/avi.c
===================================================================
--- vlc-0.8.6-svn20061012.debian.orig/modules/mux/avi.c	2007-01-08 02:28:25.000000000 +0100
+++ vlc-0.8.6-svn20061012.debian/modules/mux/avi.c	2007-01-08 02:29:13.000000000 +0100
@@ -626,8 +626,8 @@
         if( p_sys->stream[i_stream].i_duration > 0 )
         {
             i_maxbytespersec +=
-                p_sys->stream[p_sys->i_stream_video].i_totalsize /
-                p_sys->stream[p_sys->i_stream_video].i_duration;
+                p_sys->stream[i_stream].i_totalsize /
+                p_sys->stream[i_stream].i_duration;
         }
     }
 
Index: vlc-0.8.6-svn20061012.debian/modules/mux/mpeg/ts.c
===================================================================
--- vlc-0.8.6-svn20061012.debian.orig/modules/mux/mpeg/ts.c	2006-09-18 12:28:27.000000000 +0200
+++ vlc-0.8.6-svn20061012.debian/modules/mux/mpeg/ts.c	2007-01-08 02:29:13.000000000 +0100
@@ -1586,7 +1586,6 @@
             /* Select stream (lowest dts) */
             for( i = 0, i_stream = -1, i_dts = 0; i < p_mux->i_nb_inputs; i++ )
             {
-                p_input = p_mux->pp_inputs[i];
                 p_stream = (ts_stream_t*)p_mux->pp_inputs[i]->p_sys;
 
                 if( p_stream->i_pes_dts == 0 )
@@ -1606,6 +1605,7 @@
                 break;
             }
             p_stream = (ts_stream_t*)p_mux->pp_inputs[i_stream]->p_sys;
+            p_input = p_mux->pp_inputs[i_stream];
 
             /* do we need to issue pcr */
             b_pcr = VLC_FALSE;
Index: vlc-0.8.6-svn20061012.debian/modules/packetizer/mpegvideo.c
===================================================================
--- vlc-0.8.6-svn20061012.debian.orig/modules/packetizer/mpegvideo.c	2006-09-18 12:28:25.000000000 +0200
+++ vlc-0.8.6-svn20061012.debian/modules/packetizer/mpegvideo.c	2007-01-08 02:29:13.000000000 +0100
@@ -118,6 +118,7 @@
     mtime_t i_interpolated_dts;
     mtime_t i_old_duration;
     mtime_t i_last_ref_pts;
+    vlc_bool_t b_second_field;
 
     /* Number of pictures since last sequence header */
     int i_seq_old;
@@ -185,6 +186,7 @@
     p_sys->i_interpolated_dts = 0;
     p_sys->i_old_duration = 0;
     p_sys->i_last_ref_pts = 0;
+    p_sys->b_second_field = 0;
 
     p_sys->b_discontinuity = VLC_FALSE;
     p_sys->b_sync_on_intra_frame = var_CreateGetBool( p_dec, "packetizer-mpegvideo-sync-iframe" );
@@ -433,14 +435,16 @@
         else
         {
             /* Correct interpolated dts when we receive a new pts/dts */
-            if( p_sys->i_last_ref_pts > 0 )
+            if( p_sys->i_last_ref_pts > 0 && !p_sys->b_second_field )
                 p_sys->i_interpolated_dts = p_sys->i_last_ref_pts;
             if( p_sys->i_dts > 0 ) p_sys->i_interpolated_dts = p_sys->i_dts;
 
-            p_sys->i_last_ref_pts = p_sys->i_pts;
+            if( !p_sys->b_second_field )
+                p_sys->i_last_ref_pts = p_sys->i_pts;
         }
 
         p_pic->i_dts = p_sys->i_interpolated_dts;
+        p_sys->i_interpolated_dts += i_duration;
 
         /* Set PTS only if we have a B frame or if it comes from the stream */
         if( p_sys->i_pts > 0 )
@@ -456,17 +460,6 @@
             p_pic->i_pts = 0;
         }
 
-        if( p_sys->b_low_delay || p_sys->i_picture_type == 0x03 )
-        {
-            /* Trivial case (DTS == PTS) */
-            p_sys->i_interpolated_dts += i_duration;
-        }
-        else
-        {
-            p_sys->i_interpolated_dts += p_sys->i_old_duration;
-            p_sys->i_old_duration = i_duration;
-        }
-
         switch ( p_sys->i_picture_type )
         {
         case 0x01:
@@ -491,6 +484,15 @@
         p_sys->p_frame = NULL;
         p_sys->pp_last = &p_sys->p_frame;
         p_sys->b_frame_slice = VLC_FALSE;
+
+        if( p_sys->i_picture_structure != 0x03 )
+        {
+            p_sys->b_second_field = !p_sys->b_second_field;
+        }
+        else
+        {
+            p_sys->b_second_field = 0;
+        }
     }
 
     /*
Index: vlc-0.8.6-svn20061012.debian/modules/services_discovery/hal.c
===================================================================
--- vlc-0.8.6-svn20061012.debian.orig/modules/services_discovery/hal.c	2006-09-18 12:28:28.000000000 +0200
+++ vlc-0.8.6-svn20061012.debian/modules/services_discovery/hal.c	2007-01-08 02:29:14.000000000 +0100
@@ -291,11 +291,15 @@
                                                         psz_device,
                                                         "volume.disc.type" );
 #endif
-        if( !strcmp( psz_disc_type, "dvd_rom" ) )
+        if( !strncmp( psz_disc_type, "dvd_r", 5 ) )
         {
-            AddDvd( p_sd, psz_device );
+#ifdef HAVE_HAL_1
+            if (libhal_device_get_property_bool( p_sys->p_ctx, psz_device,
+                        "volume.disc.is_videodvd", NULL ) )
+#endif
+                AddDvd( p_sd, psz_device );
         }
-        else if( !strcmp( psz_disc_type, "cd_rom" ) )
+        else if( !strncmp( psz_disc_type, "cd_r", 4 ) )
         {
 #ifdef HAVE_HAL_1
             if( libhal_device_get_property_bool( p_sys->p_ctx, psz_device,
Index: vlc-0.8.6-svn20061012.debian/modules/stream_out/rtp.c
===================================================================
--- vlc-0.8.6-svn20061012.debian.orig/modules/stream_out/rtp.c	2006-10-08 18:40:54.000000000 +0200
+++ vlc-0.8.6-svn20061012.debian/modules/stream_out/rtp.c	2007-01-08 02:29:14.000000000 +0100
@@ -434,13 +434,7 @@
         sout_access_out_t *p_grab;
         char *psz_rtpmap, url[NI_MAXHOST + 8], access[17], psz_ttl[5], ipv;
 
-        if( b_rtsp )
-        {
-            msg_Err( p_stream, "muxing is not supported in RTSP mode" );
-            free( p_sys );
-            return VLC_EGENERIC;
-        }
-        else if( !p_sys->psz_destination || *p_sys->psz_destination == '\0' )
+        if( !p_sys->psz_destination || *p_sys->psz_destination == '\0' )
         {
             msg_Err( p_stream, "rtp needs a destination when muxing" );
             free( p_sys );
@@ -1642,7 +1636,7 @@
 {
     int i;
 
-    if( psz_session ) return NULL;
+    if( !psz_session ) return NULL;
 
     for( i = 0; i < p_stream->p_sys->i_rtsp; i++ )
     {
Index: vlc-0.8.6-svn20061012.debian/modules/video_output/opengl.c
===================================================================
--- vlc-0.8.6-svn20061012.debian.orig/modules/video_output/opengl.c	2006-09-18 12:28:25.000000000 +0200
+++ vlc-0.8.6-svn20061012.debian/modules/video_output/opengl.c	2007-01-08 02:29:14.000000000 +0100
@@ -224,7 +224,10 @@
     int         i_effect;
 
     float       f_speed;
+#ifdef OPENGL_MORE_EFFECT
+    // cylinder radius
     float       f_radius;
+#endif
 };
 
 /*****************************************************************************
@@ -290,7 +293,9 @@
     }
 
     p_sys->f_speed = var_CreateGetFloat( p_vout, "opengl-cube-speed" );
+#ifdef OPENGL_MORE_EFFECT
     p_sys->f_radius = var_CreateGetFloat( p_vout, "opengl-cylinder-radius" );
+#endif
 
     p_vout->pf_init = Init;
     p_vout->pf_end = End;
Index: vlc-0.8.6-svn20061012.debian/src/interface/interaction.c
===================================================================
--- vlc-0.8.6-svn20061012.debian.orig/src/interface/interaction.c	2006-09-18 12:27:56.000000000 +0200
+++ vlc-0.8.6-svn20061012.debian/src/interface/interaction.c	2007-01-08 02:29:14.000000000 +0100
@@ -441,10 +441,12 @@
 
     i_ret = intf_Interact( p_this, p_new );
 
-    if( i_ret != DIALOG_CANCELLED )
+    if( i_ret != DIALOG_CANCELLED && i_ret != VLC_EGENERIC )
     {
-        *ppsz_login = strdup( p_new->pp_widgets[1]->val.psz_string );
-        *ppsz_password = strdup( p_new->pp_widgets[2]->val.psz_string );
+        *ppsz_login = p_new->pp_widgets[1]->val.psz_string? 
+                                strdup( p_new->pp_widgets[1]->val.psz_string ) : NULL;
+        *ppsz_password = p_new->pp_widgets[2]->val.psz_string?
+                                strdup( p_new->pp_widgets[2]->val.psz_string ) : NULL;
     }
     return i_ret;
 }
Index: vlc-0.8.6-svn20061012.debian/src/stream_output/sap.c
===================================================================
--- vlc-0.8.6-svn20061012.debian.orig/src/stream_output/sap.c	2006-09-22 17:14:27.000000000 +0200
+++ vlc-0.8.6-svn20061012.debian/src/stream_output/sap.c	2007-01-08 02:29:14.000000000 +0100
@@ -76,7 +76,7 @@
 static int CalculateRate( sap_handler_t *p_sap, sap_address_t *p_address );
 static char *SDPGenerate( sap_handler_t *p_sap,
                           const session_descriptor_t *p_session,
-                          const sap_address_t *p_addr );
+                          const sap_address_t *p_addr, vlc_bool_t b_ssm );
 
 static int announce_SendSAPAnnounce( sap_handler_t *p_sap,
                                      sap_session_t *p_session );
@@ -235,7 +235,7 @@
 {
     int i_header_size, i;
     char *psz_head, psz_addr[NI_MAXNUMERICHOST];
-    vlc_bool_t b_ipv6 = VLC_FALSE;
+    vlc_bool_t b_ipv6 = VLC_FALSE, b_ssm = VLC_FALSE;
     sap_session_t *p_sap_session;
     mtime_t i_hash;
     struct addrinfo hints, *res;
@@ -288,8 +288,13 @@
             memcpy( a6->s6_addr + 2, "\x00\x00\x00\x00\x00\x00"
                    "\x00\x00\x00\x00\x00\x02\x7f\xfe", 14 );
             if( IN6_IS_ADDR_MULTICAST( a6 ) )
-                 /* force flags to zero, preserve scope */
+            {
+                /* SSM <=> ff3x::/32 */
+                b_ssm = (U32_AT (a6->s6_addr) & 0xfff0ffff) == 0xff300000;
+
+                /* force flags to zero, preserve scope */
                 a6->s6_addr[1] &= 0xf;
+            }
             else
                 /* Unicast IPv6 - assume global scope */
                 memcpy( a6->s6_addr, "\xff\x0e", 2 );
@@ -321,7 +326,11 @@
                 ipv4 = 0;
             else
             /* other addresses => 224.2.127.254 */
+            {
+                /* SSM: 232.0.0.0/8 */
+                b_ssm = (ipv4 >> 24) == 232;
                 ipv4 = 0xe0027ffe;
+            }
 
             if( ipv4 == 0 )
             {
@@ -468,7 +477,7 @@
     if( p_session->psz_sdp == NULL )
     {
         p_session->psz_sdp = SDPGenerate( p_sap, p_session,
-                                          p_sap_session->p_address );
+                                          p_sap_session->p_address, b_ssm );
         if( p_session->psz_sdp == NULL )
         {
             vlc_mutex_unlock( &p_sap->object_lock );
@@ -584,13 +593,15 @@
 
 static char *SDPGenerate( sap_handler_t *p_sap,
                           const session_descriptor_t *p_session,
-                          const sap_address_t *p_addr )
+                          const sap_address_t *p_addr, vlc_bool_t b_ssm )
 {
     int64_t i_sdp_id = mdate();
     int     i_sdp_version = 1 + p_sap->i_sessions + (rand()&0xfff);
     char *psz_group, *psz_name, psz_uribuf[NI_MAXNUMERICHOST], *psz_uri,
          *psz_sdp;
     char ipv;
+    char *sfilter = NULL;
+    int res;
 
     psz_group = p_session->psz_group;
     psz_name = p_session->psz_name;
@@ -614,30 +625,41 @@
     else
         psz_uri = p_session->psz_uri;
 
+    if (b_ssm)
+    {
+        if (asprintf (&sfilter, "a=source-filter: incl IN IP%c * %s\r\n",
+                      ipv, p_addr->psz_machine) == -1)
+            return NULL;
+    }
+
     /* see the lists in modules/stream_out/rtp.c for compliance stuff */
-    if( asprintf( &psz_sdp,
-                            "v=0\r\n"
-                            "o=- "I64Fd" %d IN IP%c %s\r\n"
-                            "s=%s\r\n"
-                            "c=IN IP%c %s/%d\r\n"
-                            "t=0 0\r\n"
-                            "a=tool:"PACKAGE_STRING"\r\n"
-                            "a=recvonly\r\n"
-                            "a=type:broadcast\n"
-                            "a=source-filter: incl IN IP%c * %s\r\n"
-                            "m=video %d %s %d\r\n"
-                            "%s%s%s",
-                            i_sdp_id, i_sdp_version,
-                            ipv, p_addr->psz_machine,
-                            psz_name, ipv, psz_uri,
-                            /* FIXME: 1 is IPv4 default TTL, not that of IPv6 */
-                            p_session->i_ttl ?: (config_GetInt( p_sap, "ttl" ) ?: 1),
-                            ipv, psz_uri,
-                            p_session->i_port, 
-                            p_session->b_rtp ? "RTP/AVP" : "udp",
-                            p_session->i_payload,
-                            psz_group ? "a=x-plgroup:" : "",
-                            psz_group ? psz_group : "", psz_group ? "\r\n" : "" ) == -1 )
+    res = asprintf (&psz_sdp,
+                        "v=0\r\n"
+                        "o=- "I64Fd" %d IN IP%c %s\r\n"
+                        "s=%s\r\n"
+                        "c=IN IP%c %s/%d\r\n"
+                        "t=0 0\r\n"
+                        "a=tool:"PACKAGE_STRING"\r\n"
+                        "a=recvonly\r\n"
+                        "a=type:broadcast\n"
+                        "%s"
+                        "m=video %d %s %d\r\n"
+                        "%s%s%s",
+                        i_sdp_id, i_sdp_version,
+                        ipv, p_addr->psz_machine,
+                        psz_name, ipv, psz_uri,
+                        /* FIXME: 1 is IPv4 default TTL, not that of IPv6 */
+                        p_session->i_ttl ?: (config_GetInt( p_sap, "ttl" ) ?: 1),
+                        (sfilter != NULL) ? sfilter : "",
+                        p_session->i_port,
+                        p_session->b_rtp ? "RTP/AVP" : "udp",
+                        p_session->i_payload,
+                        psz_group ? "a=x-plgroup:" : "",
+                        psz_group ? psz_group : "", psz_group ? "\r\n" : "");
+    if (sfilter != NULL)
+        free (sfilter);
+
+    if (res == -1)
         return NULL;
 
     msg_Dbg( p_sap, "Generated SDP (%i bytes):\n%s", strlen(psz_sdp),
