diff -puriN vlc-0.8.6/share/http/js/vlm.js vlc-0.8.6a/share/http/js/vlm.js
--- vlc-0.8.6/share/http/js/vlm.js	2006-09-18 12:28:00.000000000 +0200
+++ vlc-0.8.6a/share/http/js/vlm.js	2007-01-07 23:06:40.000000000 +0100
@@ -153,7 +153,7 @@ function vlm_schedule_type_change( name 
 
 function sanitize_input( str )
 {
-    return str.replace( /\\/g, '\\\\').replace( /\'/g, '\\\'' ).replace( /\"/g, '\\\"' ).replace( /^/, '"' ).replace( /$/, '"' ).replace( /\ option\ /g, '" option "' );
+    return str.replace( /\"/g, '\\\"' ).replace( /^/, '"' ).replace( /$/, '"' ).replace( /\ option\ /g, '" option "' );
 }
 
 function update_vlm_add_broadcast()
