diff -Nurad vlc-0.8.6.c.orig/modules/demux/playlist/m3u.c vlc-0.8.6.c/modules/demux/playlist/m3u.c
--- vlc-0.8.6.c.orig/modules/demux/playlist/m3u.c	2008-01-11 14:30:15.000000000 +0100
+++ vlc-0.8.6.c/modules/demux/playlist/m3u.c	2008-01-11 14:49:35.000000000 +0100
@@ -137,6 +137,8 @@
     mtime_t    i_duration = -1;
     char       **ppsz_options = NULL;
     int        i_options = 0, i;
+    vlc_bool_t b_enable_extvlcopt = config_GetInt( p_demux, "m3u-extvlcopt" );
+
 
     playlist_item_t *p_item, *p_current;
 
@@ -188,20 +190,28 @@
                     psz_name = strdup( psz_name );
                 if ( psz_artist )
                     psz_artist = strdup( psz_artist );
-            }
             else if( !strncasecmp( psz_parse, "EXTVLCOPT:",
                                    sizeof("EXTVLCOPT:") -1 ) )
             {
-                /* VLC Option */
-                char *psz_option;
-                psz_parse += sizeof("EXTVLCOPT:") -1;
-                if( !*psz_parse ) goto error;
+                if( b_enable_extvlcopt )
+                {
+                    /* VLC Option */
+                    char *psz_option;
+                    psz_parse += sizeof("EXTVLCOPT:") -1;
+                    if( !*psz_parse ) goto error;
 
-                psz_option = MaybeFromLocaleDup( psz_parse );
-                if( psz_option )
-                    INSERT_ELEM( ppsz_options, i_options, i_options,
-                                 psz_option );
+                    psz_option = MaybeFromLocaleDup( psz_parse );
+                    if( psz_option )
+                        INSERT_ELEM( ppsz_options, i_options, i_options,
+                                     psz_option );
+                }
+                else
+                {
+                    msg_Err( p_demux, "m3u EXTVLCOPT parsing is disabled for security reasons. If you need it and trust the m3u playlist you are trying to open, please append --m3u-extvlcopt to your command line." );
+                }
+            }
             }
+
         }
         else if( *psz_parse )
         {
--- vlc-0.8.6.c.orig/modules/demux/playlist/playlist.c	2008-01-11 14:30:14.000000000 +0100
+++ vlc-0.8.6.c/modules/demux/playlist/playlist.c	2008-01-11 14:49:35.000000000 +0100
@@ -42,6 +42,11 @@
 #define SHOW_ADULT_LONGTEXT N_( "Show NC17 rated video streams when " \
                 "using shoutcast video playlists." )
 
+#define EXTVLCOPT_TEXT N_( "Enable parsing of EXTVLCOPT: options" )
+#define EXTVLCOPT_LONGTEXT N_( "Enable parsing of EXTVLCOPT: options in m3u " \
+    "playlists. This option is default disabled to prevent untrusted sources " \
+    "using VLC options without the user's knowledge." )
+
 vlc_module_begin();
     add_shortcut( "playlist" );
     set_category( CAT_INPUT );
@@ -67,6 +72,8 @@
         set_description( _("M3U playlist import") );
         add_shortcut( "m3u-open" );
         set_capability( "demux2", 10 );
+        add_bool( "m3u-extvlcopt", VLC_FALSE, NULL,
+                  EXTVLCOPT_TEXT, EXTVLCOPT_LONGTEXT, VLC_FALSE );
         set_callbacks( E_(Import_M3U), E_(Close_M3U) );
     add_submodule();
         set_description( _("PLS playlist import") );
