diff -Nurad vlc-0.8.6.c.orig/src/network/httpd.c vlc-0.8.6.c/src/network/httpd.c
--- vlc-0.8.6.c.orig/src/network/httpd.c	2008-01-03 20:18:35.000000000 +0100
+++ vlc-0.8.6.c/src/network/httpd.c	2008-01-03 20:21:06.000000000 +0100
@@ -414,7 +414,7 @@
     psz_connection = httpd_MsgGet( &cl->query, "Connection" );
     if( psz_connection != NULL )
     {
-        httpd_MsgAdd( answer, "Connection", psz_connection );
+        httpd_MsgAdd( answer, "Connection", "%s", psz_connection );
     }
 
     httpd_MsgAdd( answer, "Content-Length", "%d", answer->i_body );
