Description: Fix v4l2 build failure on kfreebsd.
Author: Benjamin Drung <bdrung@debian.org>
Bug-Debian: http://bugs.debian.org/660935

--- a/modules/access/v4l2/v4l2.h
+++ b/modules/access/v4l2/v4l2.h
@@ -33,11 +33,8 @@
 # warning Please update Video4Linux2 headers!
 # define V4L2_CTRL_FLAG_VOLATILE 0x0080
 #endif
-#ifdef __linux__
-# include <linux/version.h>
-# if LINUX_VERSION_CODE < KERNEL_VERSION(3,1,0)
-#  define V4L2_CTRL_TYPE_BITMASK 8
-# endif
+#ifndef V4L2_CTRL_TYPE_BITMASK /* 3.1 */
+# define V4L2_CTRL_TYPE_BITMASK 8
 #endif
 #ifndef V4L2_CID_ILLUMINATORS_1 /* 2.6.37 */
 # define V4L2_CID_ILLUMINATORS_1 (V4L2_CID_BASE+38)
