Description: Make vlc compatible with xulrunner 1.9.1
Author: Benjamin Drung <bdrung@debian.org>
Forwarded: not-needed

--- a/configure
+++ b/configure
@@ -47504,9 +47504,9 @@
           if test -n "$i"
           then
             echo "Trying to find $i package" >&5
-            { $as_echo "$as_me:${as_lineno-$LINENO}: checking for $i >= 1.9.2" >&5
-$as_echo_n "checking for $i >= 1.9.2... " >&6; }
-            if $PKG_CONFIG --exists --print-errors "$i >= 1.9.2" 2>&5
+            { $as_echo "$as_me:${as_lineno-$LINENO}: checking for $i >= 1.9.1" >&5
+$as_echo_n "checking for $i >= 1.9.1... " >&6; }
+            if $PKG_CONFIG --exists --print-errors "$i >= 1.9.1" 2>&5
             then
               echo "Using $i pkg-config package." >&5
               echo "Using $i package." >&6
@@ -47734,7 +47734,7 @@
 done
 test -n "$MOZILLA_CONFIG" || MOZILLA_CONFIG="no"
 
-            test "${MOZILLA_CONFIG}" = "no" && as_fn_error $? "Please install the Mozilla development tools (version 1.9.2 or higher) or use --disable-mozilla." "$LINENO" 5
+            test "${MOZILLA_CONFIG}" = "no" && as_fn_error $? "Please install the Mozilla development tools (version 1.9.1 or higher) or use --disable-mozilla." "$LINENO" 5
 
 
 fi
--- a/configure.ac
+++ b/configure.ac
@@ -4556,8 +4556,8 @@
           if test -n "$i"
           then
             echo "Trying to find $i package" >&AS_MESSAGE_LOG_FD
-            AC_MSG_CHECKING([for $i >= 1.9.2])
-            if $PKG_CONFIG --exists --print-errors "$i >= 1.9.2" 2>&AS_MESSAGE_LOG_FD
+            AC_MSG_CHECKING([for $i >= 1.9.1])
+            if $PKG_CONFIG --exists --print-errors "$i >= 1.9.1" 2>&AS_MESSAGE_LOG_FD
             then
               echo "Using $i pkg-config package." >&AS_MESSAGE_LOG_FD
               echo "Using $i package." >&AS_MESSAGE_FD
--- a/projects/mozilla/vlcplugin.h
+++ b/projects/mozilla/vlcplugin.h
@@ -29,6 +29,10 @@
 #ifndef __VLCPLUGIN_H__
 #define __VLCPLUGIN_H__
 
+// Changes for compatibility to xulrunner 1.9.1
+#define UTF8Characters utf8characters
+#define UTF8Length utf8length
+
 #include <vlc/vlc.h>
 
 // Setup XP_MACOSX, XP_UNIX, XP_WIN
--- a/projects/mozilla/control/nporuntime.cpp
+++ b/projects/mozilla/control/nporuntime.cpp
@@ -27,8 +27,8 @@
 #include <string.h>
 #include <stdlib.h>
 
-#include "nporuntime.h"
 #include "vlcplugin.h"
+#include "nporuntime.h"
 
 char* RuntimeNPObject::stringValue(const NPString &s)
 {
