Description: Make vlc compatible with xulrunner 1.9.1
Author: Benjamin Drung <bdrung@debian.org>
Forwarded: not-needed

--- a/configure.ac
+++ b/configure.ac
@@ -4559,8 +4559,8 @@
           if test -n "$i"
           then
             echo "Trying to find $i package" >&AS_MESSAGE_LOG_FD
-            AC_MSG_CHECKING([for $i >= 1.9.2])
-            if $PKG_CONFIG --exists --print-errors "$i >= 1.9.2" 2>&AS_MESSAGE_LOG_FD
+            AC_MSG_CHECKING([for $i >= 1.9.1])
+            if $PKG_CONFIG --exists --print-errors "$i >= 1.9.1" 2>&AS_MESSAGE_LOG_FD
             then
               echo "Using $i pkg-config package." >&AS_MESSAGE_LOG_FD
               echo "Using $i package." >&AS_MESSAGE_FD
--- a/projects/mozilla/vlcplugin.h
+++ b/projects/mozilla/vlcplugin.h
@@ -29,6 +29,10 @@
 #ifndef __VLCPLUGIN_H__
 #define __VLCPLUGIN_H__
 
+// Changes for compatibility to xulrunner 1.9.1
+#define UTF8Characters utf8characters
+#define UTF8Length utf8length
+
 #include <vlc/vlc.h>
 
 // Setup XP_MACOSX, XP_UNIX, XP_WIN
--- a/projects/mozilla/control/nporuntime.cpp
+++ b/projects/mozilla/control/nporuntime.cpp
@@ -27,8 +27,8 @@
 #include <string.h>
 #include <stdlib.h>
 
-#include "nporuntime.h"
 #include "vlcplugin.h"
+#include "nporuntime.h"
 
 char* RuntimeNPObject::stringValue(const NPString &s)
 {
