--- x265/source/encoder/ratecontrol.cpp.orig	2016-12-07 11:36:58.694172335 +0100
+++ x265/source/encoder/ratecontrol.cpp	2016-12-07 11:37:22.990215303 +0100
@@ -48,7 +48,7 @@
 {\
     bErr = 0;\
     p = strstr(opts, opt "=");\
-    char* q = strstr(opts, "no-"opt);\
+    char* q = strstr(opts, "no-" opt);\
     if (p && sscanf(p, opt "=%d" , &i) && param_val != i)\
         bErr = 1;\
     else if (!param_val && !q)\
--- x265/source/common/param.cpp.orig	2016-12-07 11:36:10.430085438 +0100
+++ x265/source/common/param.cpp	2016-12-07 11:36:32.238124965 +0100
@@ -1196,7 +1196,7 @@
         return NULL;
 
 #define BOOL(param, cliopt) \
-    s += sprintf(s, " %s", (param) ? cliopt : "no-"cliopt);
+    s += sprintf(s, " %s", (param) ? cliopt : "no-" cliopt);
 
     s += sprintf(s, "%dx%d", p->sourceWidth,p->sourceHeight);
     s += sprintf(s, " fps=%u/%u", p->fpsNum, p->fpsDenom);
