Index: Makefile
===================================================================
--- Makefile	(revision 7407)
+++ Makefile	(working copy)
@@ -159,7 +159,9 @@
 ifeq ($(CONFIG_PP),yes)
 	$(MAKE) -C libpostproc install-headers
 endif
+ifeq ($(CONFIG_SWSCALER),yes)
 	$(MAKE) -C libswscale  install-headers
+endif
 
 uninstall: uninstall-progs uninstall-libs uninstall-headers uninstall-man uninstall-vhook
 
