--- libgcrypt/Makefile.am.orig	2007-05-04 12:11:28.000000000 +0200
+++ libgcrypt/Makefile.am	2008-05-24 00:44:08.000000000 +0200
@@ -23,7 +23,7 @@
   --enable-ciphers=arcfour:blowfish:cast5:des:aes:twofish:serpent:rfc2268:seed:camellia
 
 DIST_SUBDIRS = m4 mpi cipher src doc tests
-SUBDIRS = mpi cipher src doc tests
+SUBDIRS = mpi cipher src
 EXTRA_DIST = BUGS autogen.sh README.SVN
 DISTCLEANFILES = 
 
