--- src/dvd_reader.c	2008-09-11 14:14:43.000000000 +0200
+++ src/dvd_reader.c	2008-09-11 14:18:53.000000000 +0200
@@ -32,6 +32,7 @@
 #include <limits.h>
 #include <dirent.h>
 
+#if 0
 /* misc win32 helpers */
 #ifdef WIN32
 #ifndef HAVE_GETTIMEOFDAY
@@ -50,6 +51,7 @@
 #include <io.h> /* read() */
 #define lseek64 _lseeki64
 #endif
+#endif
 
 #if defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__NetBSD__) || defined(__bsdi__)|| defined(__DARWIN__)
 #define SYS_BSD 1
